diff --git a/substrate/client/network/src/request_responses.rs b/substrate/client/network/src/request_responses.rs
index 68ae8e0..c02c275 100644
--- a/substrate/client/network/src/request_responses.rs
+++ b/substrate/client/network/src/request_responses.rs
@@ -757,6 +757,7 @@ impl NetworkBehaviour for RequestResponsesBehaviour {
 						request_response::Event::Message {
 							peer,
 							message: Message::Request { request_id, request, channel, .. },
+							..
 						} => {
 							self.pending_responses_arrival_time
 								.insert((protocol.clone(), request_id).into(), Instant::now());
@@ -957,7 +958,7 @@ impl NetworkBehaviour for RequestResponsesBehaviour {
 						},
 
 						// A response to an inbound request has been sent.
-						request_response::Event::ResponseSent { request_id, peer } => {
+						request_response::Event::ResponseSent { request_id, peer, .. } => {
 							let arrival_time = self
 								.pending_responses_arrival_time
 								.remove(&(protocol.clone(), request_id).into())
diff --git a/substrate/client/network/src/behaviour.rs b/substrate/client/network/src/behaviour.rs
index 0f6b1ab..0310978 100644
--- a/substrate/client/network/src/behaviour.rs
+++ b/substrate/client/network/src/behaviour.rs
@@ -422,8 +422,8 @@ impl From<DiscoveryOut> for BehaviourOut {
 	}
 }
 
-impl From<void::Void> for BehaviourOut {
-	fn from(e: void::Void) -> Self {
-		void::unreachable(e)
+impl From<core::convert::Infallible> for BehaviourOut {
+	fn from(_: core::convert::Infallible) -> Self {
+		unreachable!()
 	}
 }
diff --git a/substrate/client/network/src/service.rs b/substrate/client/network/src/service.rs
index 26a16a5..068a149 100644
--- a/substrate/client/network/src/service.rs
+++ b/substrate/client/network/src/service.rs
@@ -113,17 +113,36 @@ pub mod traits;
 const LOG_TARGET: &str = "sub-libp2p";
 
 struct Libp2pBandwidthSink {
-	#[allow(deprecated)]
-	sink: Arc<transport::BandwidthSinks>,
+	sink: libp2p::metrics::Registry,
 }
 
+// A horrific implementation of the API to resolve existing API boundaries
+use core::str::FromStr;
 impl BandwidthSink for Libp2pBandwidthSink {
 	fn total_inbound(&self) -> u64 {
-		self.sink.total_inbound()
+		let mut buf = String::new();
+		prometheus_client::encoding::text::encode(&mut buf, &self.sink).unwrap();
+
+		let mut total = 0u64;
+		for line in buf.lines().filter(|line| !line.starts_with('#')).filter(|line| line.to_lowercase().contains("inbound")) {
+		  if let Some(part) = line.split(' ').skip(1).next() {
+			total = total.saturating_add(u64::from_str(part).unwrap_or(u64::MAX));
+		  }
+		}
+		total
 	}
 
 	fn total_outbound(&self) -> u64 {
-		self.sink.total_outbound()
+		let mut buf = String::new();
+		prometheus_client::encoding::text::encode(&mut buf, &self.sink).unwrap();
+
+		let mut total = 0u64;
+		for line in buf.lines().filter(|line| !line.starts_with('#')).filter(|line| line.to_lowercase().contains("outbound")) {
+		  if let Some(part) = line.split(' ').skip(1).next() {
+			total = total.saturating_add(u64::from_str(part).unwrap_or(u64::MAX));
+		  }
+		}
+		total
 	}
 }
 
@@ -1746,12 +1765,7 @@ where
 					if let Some(addresses) =
 						not_reported.then(|| self.boot_node_ids.get(&peer_id)).flatten()
 					{
-						if let DialError::WrongPeerId { obtained, endpoint } = &error {
-							if let ConnectedPoint::Dialer {
-								address,
-								role_override: _,
-								port_use: _,
-							} = endpoint
+						if let DialError::WrongPeerId { obtained, address } = &error {
 							{
 								let address_without_peer_id = parse_addr(address.clone().into())
 									.map_or_else(|_| address.clone(), |r| r.1.into());
@@ -1806,6 +1820,7 @@ where
 				local_addr,
 				send_back_addr,
 				error,
+				peer_id: _,
 			} => {
 				debug!(
 					target: LOG_TARGET,
diff --git a/substrate/client/network/src/transport.rs b/substrate/client/network/src/transport.rs
index 2f6b7a6..61ba35b 100644
--- a/substrate/client/network/src/transport.rs
+++ b/substrate/client/network/src/transport.rs
@@ -25,14 +25,10 @@ use libp2p::{
 		transport::{Boxed, OptionalTransport},
 		upgrade,
 	},
-	dns, identity, noise, tcp, websocket, PeerId, Transport, TransportExt,
+	dns, identity, noise, tcp, websocket, PeerId, Transport,
 };
 use std::{sync::Arc, time::Duration};
 
-// TODO: Create a wrapper similar to upstream `BandwidthTransport` that tracks sent/received bytes
-#[allow(deprecated)]
-pub use libp2p::bandwidth::BandwidthSinks;
-
 /// Builds the transport that serves as a common ground for all connections.
 ///
 /// If `memory_only` is true, then only communication within the same process are allowed. Only
@@ -40,11 +36,10 @@ pub use libp2p::bandwidth::BandwidthSinks;
 ///
 /// Returns a `BandwidthSinks` object that allows querying the average bandwidth produced by all
 /// the connections spawned with this transport.
-#[allow(deprecated)]
-pub fn build_transport(
+pub(crate) fn build_transport(
 	keypair: identity::Keypair,
 	memory_only: bool,
-) -> (Boxed<(PeerId, StreamMuxerBox)>, Arc<BandwidthSinks>) {
+) -> (Boxed<(PeerId, StreamMuxerBox)>, libp2p::metrics::Registry) {
 	// Build the base layer of the transport.
 	let transport = if !memory_only {
 		// Main transport: DNS(TCP)
@@ -76,12 +71,12 @@ pub fn build_transport(
 	let authentication_config = noise::Config::new(&keypair).expect("Can create noise config. qed");
 	let multiplexing_config = libp2p::yamux::Config::default();
 
+	let mut registry = libp2p::metrics::Registry::default();
 	let transport = transport
 		.upgrade(upgrade::Version::V1Lazy)
 		.authenticate(authentication_config)
 		.multiplex(multiplexing_config)
-		.timeout(Duration::from_secs(20))
-		.boxed();
-
-	transport.with_bandwidth_logging()
+		.timeout(Duration::from_secs(20));
+	let transport = libp2p::metrics::BandwidthTransport::new(transport, &mut registry);
+	(transport.map(|(peer_id, conn), _| (peer_id, StreamMuxerBox::new(conn))).boxed(), registry)
 }
diff --git a/substrate/client/network/Cargo.toml b/substrate/client/network/Cargo.toml
index 1f40ea9..5087611 100644
--- a/substrate/client/network/Cargo.toml
+++ b/substrate/client/network/Cargo.toml
@@ -32,5 +32,6 @@ fnv = { workspace = true }
 futures = { workspace = true }
 futures-timer = { workspace = true }
 ip_network = { workspace = true }
-libp2p = { features = ["dns", "identify", "kad", "macros", "mdns", "noise", "ping", "request-response", "tcp", "tokio", "websocket", "yamux"], workspace = true }
+libp2p = { features = ["dns", "identify", "kad", "macros", "mdns", "noise", "ping", "request-response", "tcp", "tokio", "websocket", "yamux", "metrics"], workspace = true }
+prometheus-client = "0.23"
 linked_hash_set = { workspace = true }
