diff --git a/substrate/client/executor/wasmtime/Cargo.toml b/substrate/client/executor/wasmtime/Cargo.toml
index d6ab3e13..cc080d8b 100644
--- a/substrate/client/executor/wasmtime/Cargo.toml
+++ b/substrate/client/executor/wasmtime/Cargo.toml
@@ -31,26 +31,22 @@ sc-executor-common.workspace = true
 sp-runtime-interface.default-features = true
 sp-runtime-interface.workspace = true
 sp-wasm-interface = { features = ["wasmtime"], workspace = true, default-features = true }
-wasmtime = { features = [
+wasmtime = { default-features = false, features = [
 	"addr2line",
 	"cache",
 	"cranelift",
-	"demangle",
+	# "demangle",
-	"gc",
-	"gc-null",
+	# "gc",
+	# "gc-null",
 	"parallel-compilation",
 	"pooling-allocator",
 	"profiling",
-	"threads",
+	# "threads",
 ], workspace = true }
 
 # Here we include the rustix crate in the exactly same semver-compatible version as used by
-# wasmtime and enable its 'use-libc' flag.
-#
-# By default rustix directly calls the appropriate syscalls completely bypassing libc;
-# this doesn't have any actual benefits for us besides making it harder to debug memory
-# problems (since then `mmap` etc. cannot be easily hooked into).
-rustix = { features = ["fs", "mm", "param", "std", "use-libc"], workspace = true }
+# wasmtime
+rustix = { features = ["fs", "mm", "param", "std"], workspace = true }
 
 [dev-dependencies]
 cargo_metadata = { workspace = true }
diff --git a/substrate/client/executor/wasmtime/src/runtime.rs b/substrate/client/executor/wasmtime/src/runtime.rs
index fd2b2f9e..2e8fc2fa 100644
--- a/substrate/client/executor/wasmtime/src/runtime.rs
+++ b/substrate/client/executor/wasmtime/src/runtime.rs
@@ -217,8 +217,8 @@ fn setup_wasmtime_caching(
 
 fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config, WasmError> {
 	let mut config = wasmtime::Config::new();
-	config.cranelift_opt_level(wasmtime::OptLevel::SpeedAndSize);
-	config.cranelift_nan_canonicalization(semantics.canonicalize_nans);
+	config.cranelift_opt_level(wasmtime::OptLevel::Speed);
+	config.cranelift_pcc(false);
 
 	let profiler = match std::env::var_os("WASMTIME_PROFILING_STRATEGY") {
 		Some(os_string) if os_string == "jitdump" => wasmtime::ProfilingStrategy::JitDump,
@@ -251,16 +251,16 @@ fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config,
 
 	// Be clear and specific about the extensions we support. If an update brings new features
 	// they should be introduced here as well.
-	config.wasm_reference_types(semantics.wasm_reference_types);
-	config.wasm_simd(semantics.wasm_simd);
-	config.wasm_relaxed_simd(semantics.wasm_simd);
-	config.wasm_bulk_memory(semantics.wasm_bulk_memory);
-	config.wasm_multi_value(semantics.wasm_multi_value);
-	config.wasm_multi_memory(false);
-	config.wasm_threads(false);
-	config.wasm_memory64(false);
-	config.wasm_tail_call(false);
-	config.wasm_extended_const(false);
+	let mut features = wasmtime::WasmFeatures::empty();
+	features.insert(wasmtime::WasmFeatures::FLOATS);
+	config.cranelift_nan_canonicalization(true);
+	features.insert(wasmtime::WasmFeatures::MUTABLE_GLOBAL);
+	features.set(wasmtime::WasmFeatures::MULTI_VALUE, semantics.wasm_multi_value);
+	features.set(wasmtime::WasmFeatures::BULK_MEMORY, semantics.wasm_bulk_memory);
+	features.set(wasmtime::WasmFeatures::REFERENCE_TYPES, semantics.wasm_reference_types);
+	features.set(wasmtime::WasmFeatures::SIMD, semantics.wasm_simd);
+	config.wasm_features(wasmtime::WasmFeatures::all(), false);
+	config.wasm_features(features, true);
 
 	let (use_pooling, use_cow) = match semantics.instantiation_strategy {
 		InstantiationStrategy::PoolingCopyOnWrite => (true, true),
@@ -408,7 +408,7 @@ pub struct Semantics {
 	// I.e. if [`CodeSupplyMode::Verbatim`] is used.
 	pub deterministic_stack_limit: Option<DeterministicStackLimit>,
 
-	/// Controls whether wasmtime should compile floating point in a way that doesn't allow for
+	/* /// Controls whether wasmtime should compile floating point in a way that doesn't allow for
 	/// non-determinism.
 	///
 	/// By default, the wasm spec allows some local non-determinism wrt. certain floating point
@@ -420,7 +420,7 @@ pub struct Semantics {
 	/// The classical runtime environment for Substrate allowed it and punted this on the runtime
 	/// developers. For PVFs, we want to ensure that execution is deterministic though. Therefore,
 	/// for PVF execution this flag is meant to be turned on.
-	pub canonicalize_nans: bool,
+	pub canonicalize_nans: bool, */
 
 	/// Configures wasmtime to use multiple threads for compiling.
 	pub parallel_compilation: bool,
diff --git a/substrate/client/executor/src/wasm_runtime.rs b/substrate/client/executor/src/wasm_runtime.rs
index 8f189ca9..280eb1d2 100644
--- a/substrate/client/executor/src/wasm_runtime.rs
+++ b/substrate/client/executor/src/wasm_runtime.rs
@@ -311,7 +311,7 @@ where
 						heap_alloc_strategy,
 						instantiation_strategy,
 						deterministic_stack_limit: None,
-						canonicalize_nans: false,
+						// canonicalize_nans: false,
 						parallel_compilation: true,
 						wasm_multi_value: false,
 						wasm_bulk_memory: false,
