diff --git a/substrate/frame/support/procedural/src/construct_runtime/expand/mod.rs b/substrate/frame/support/procedural/src/construct_runtime/expand/mod.rs
index 823aa69..1d98cfa 100644
--- a/substrate/frame/support/procedural/src/construct_runtime/expand/mod.rs
+++ b/substrate/frame/support/procedural/src/construct_runtime/expand/mod.rs
@@ -22,7 +22,6 @@ mod freeze_reason;
 mod hold_reason;
 mod inherent;
 mod lock_id;
-mod metadata;
 mod origin;
 mod outer_enums;
 mod slash_reason;
@@ -36,7 +35,6 @@ pub use freeze_reason::expand_outer_freeze_reason;
 pub use hold_reason::expand_outer_hold_reason;
 pub use inherent::expand_outer_inherent;
 pub use lock_id::expand_outer_lock_id;
-pub use metadata::expand_runtime_metadata;
 pub use origin::expand_outer_origin;
 pub use outer_enums::{expand_outer_enum, OuterEnumType};
 pub use slash_reason::expand_outer_slash_reason;
diff --git a/substrate/frame/support/procedural/src/construct_runtime/mod.rs b/substrate/frame/support/procedural/src/construct_runtime/mod.rs
index c6018e0..e433b42 100644
--- a/substrate/frame/support/procedural/src/construct_runtime/mod.rs
+++ b/substrate/frame/support/procedural/src/construct_runtime/mod.rs
@@ -401,13 +401,6 @@ fn construct_runtime_final_expansion(
 	let dispatch = expand::expand_outer_dispatch(&name, system_pallet, &pallets, &scrate);
 	let tasks = expand::expand_outer_task(&name, &pallets, &scrate);
 	let query = expand::expand_outer_query(&name, &pallets, &scrate);
-	let metadata = expand::expand_runtime_metadata(
-		&name,
-		&pallets,
-		&scrate,
-		&unchecked_extrinsic,
-		&system_pallet.path,
-	);
 	let outer_config = expand::expand_outer_config(&name, &pallets, &scrate);
 	let inherent =
 		expand::expand_outer_inherent(&name, &block, &unchecked_extrinsic, &pallets, &scrate);
@@ -495,8 +488,6 @@ fn construct_runtime_final_expansion(
 
 		#query
 
-		#metadata
-
 		#outer_config
 
 		#inherent
diff --git a/substrate/frame/support/procedural/src/runtime/expand/mod.rs b/substrate/frame/support/procedural/src/runtime/expand/mod.rs
index 005b109..5508cda 100644
--- a/substrate/frame/support/procedural/src/runtime/expand/mod.rs
+++ b/substrate/frame/support/procedural/src/runtime/expand/mod.rs
@@ -234,13 +234,6 @@ fn construct_runtime_final_expansion(
 	let all_pallets = decl_all_pallets(&name, pallets.iter(), &features);
 	let pallet_to_index = decl_pallet_runtime_setup(&name, &pallets, &scrate);
 
-	let metadata = expand::expand_runtime_metadata(
-		&name,
-		&pallets,
-		&scrate,
-		&unchecked_extrinsic,
-		&system_pallet.path,
-	);
 	let outer_config = expand::expand_outer_config(&name, &pallets, &scrate);
 	let inherent =
 		expand::expand_outer_inherent(&name, &block, &unchecked_extrinsic, &pallets, &scrate);
@@ -307,8 +300,6 @@ fn construct_runtime_final_expansion(
 
 		#query
 
-		#metadata
-
 		#outer_config
 
 		#inherent
diff --git a/substrate/test-utils/runtime/src/lib.rs b/substrate/test-utils/runtime/src/lib.rs
index 5636007..828d749 100644
--- a/substrate/test-utils/runtime/src/lib.rs
+++ b/substrate/test-utils/runtime/src/lib.rs
@@ -487,7 +487,7 @@ impl_runtime_apis! {
 
 	impl sp_api::Metadata<Block> for Runtime {
 		fn metadata() -> OpaqueMetadata {
-			OpaqueMetadata::new(Runtime::metadata().into())
+			OpaqueMetadata::new(vec![])
 		}
 
 		fn metadata_at_version(version: u32) -> Option<OpaqueMetadata> {
