diff --git a/substrate/frame/support/src/storage/types/counted_map.rs b/substrate/frame/support/src/storage/types/counted_map.rs
index ae155ea..6bfc1d1 100644
--- a/substrate/frame/support/src/storage/types/counted_map.rs
+++ b/substrate/frame/support/src/storage/types/counted_map.rs
@@ -502,8 +502,8 @@ impl<Prefix, Hasher, Key, Value, QueryKind, OnEmpty, MaxValues> StorageEntryMeta
 where
 	Prefix: CountedStorageMapInstance,
 	Hasher: crate::hash::StorageHasher,
-	Key: FullCodec + scale_info::StaticTypeInfo,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Key: FullCodec,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
diff --git a/substrate/frame/support/src/storage/types/counted_nmap.rs b/substrate/frame/support/src/storage/types/counted_nmap.rs
index 45bd82f..4abe2f8 100644
--- a/substrate/frame/support/src/storage/types/counted_nmap.rs
+++ b/substrate/frame/support/src/storage/types/counted_nmap.rs
@@ -627,7 +627,7 @@ impl<Prefix, Key, Value, QueryKind, OnEmpty, MaxValues> StorageEntryMetadataBuil
 where
 	Prefix: CountedStorageNMapInstance,
 	Key: super::key::KeyGenerator,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
diff --git a/substrate/frame/support/src/storage/types/double_map.rs b/substrate/frame/support/src/storage/types/double_map.rs
index 24aad3d..fa13a52 100644
--- a/substrate/frame/support/src/storage/types/double_map.rs
+++ b/substrate/frame/support/src/storage/types/double_map.rs
@@ -726,9 +726,9 @@ where
 	Prefix: StorageInstance,
 	Hasher1: crate::hash::StorageHasher,
 	Hasher2: crate::hash::StorageHasher,
-	Key1: FullCodec + scale_info::StaticTypeInfo,
-	Key2: FullCodec + scale_info::StaticTypeInfo,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Key1: FullCodec,
+	Key2: FullCodec,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
@@ -745,8 +745,8 @@ where
 			modifier: QueryKind::METADATA,
 			ty: StorageEntryTypeIR::Map {
 				hashers: vec![Hasher1::METADATA, Hasher2::METADATA],
-				key: scale_info::meta_type::<(Key1, Key2)>(),
-				value: scale_info::meta_type::<Value>(),
+				key: scale_info::meta_type::<()>(),
+				value: scale_info::meta_type::<()>(),
 			},
 			default: OnEmpty::get().encode(),
 			docs,
diff --git a/substrate/frame/support/src/storage/types/key.rs b/substrate/frame/support/src/storage/types/key.rs
index b9b497b..c96e55a 100755
--- a/substrate/frame/support/src/storage/types/key.rs
+++ b/substrate/frame/support/src/storage/types/key.rs
@@ -21,7 +21,6 @@ use crate::hash::{ReversibleStorageHasher, StorageHasher};
 use alloc::{boxed::Box, vec::Vec};
 use codec::{Encode, EncodeLike, FullCodec, MaxEncodedLen};
 use paste::paste;
-use scale_info::StaticTypeInfo;
 
 /// A type used exclusively by storage maps as their key type.
 ///
@@ -36,7 +35,7 @@ pub struct Key<Hasher, KeyType>(core::marker::PhantomData<(Hasher, KeyType)>);
 
 /// A trait that contains the current key as an associated type.
 pub trait KeyGenerator {
-	type Key: EncodeLike<Self::Key> + StaticTypeInfo;
+	type Key: EncodeLike<Self::Key>;
 	type KArg: Encode + EncodeLike<Self::KArg>;
 	type HashFn: FnOnce(&[u8]) -> Vec<u8>;
 	type HArg;
@@ -68,7 +67,7 @@ pub trait KeyGeneratorInner: KeyGenerator {
 	fn final_hash(encoded: &[u8]) -> Vec<u8>;
 }
 
-impl<H: StorageHasher, K: FullCodec + StaticTypeInfo> KeyGenerator for Key<H, K> {
+impl<H: StorageHasher, K: FullCodec> KeyGenerator for Key<H, K> {
 	type Key = K;
 	type KArg = (K,);
 	type HashFn = Box<dyn FnOnce(&[u8]) -> Vec<u8>>;
@@ -90,7 +89,7 @@ impl<H: StorageHasher, K: FullCodec + StaticTypeInfo> KeyGenerator for Key<H, K>
 	}
 }
 
-impl<H: StorageHasher, K: FullCodec + MaxEncodedLen + StaticTypeInfo> KeyGeneratorMaxEncodedLen
+impl<H: StorageHasher, K: FullCodec + MaxEncodedLen> KeyGeneratorMaxEncodedLen
 	for Key<H, K>
 {
 	fn key_max_encoded_len() -> usize {
@@ -98,7 +97,7 @@ impl<H: StorageHasher, K: FullCodec + MaxEncodedLen + StaticTypeInfo> KeyGenerat
 	}
 }
 
-impl<H: StorageHasher, K: FullCodec + StaticTypeInfo> KeyGeneratorInner for Key<H, K> {
+impl<H: StorageHasher, K: FullCodec> KeyGeneratorInner for Key<H, K> {
 	type Hasher = H;
 
 	fn final_hash(encoded: &[u8]) -> Vec<u8> {
@@ -235,7 +234,7 @@ pub trait ReversibleKeyGenerator: KeyGenerator {
 	fn decode_final_key(key_material: &[u8]) -> Result<(Self::Key, &[u8]), codec::Error>;
 }
 
-impl<H: ReversibleStorageHasher, K: FullCodec + StaticTypeInfo> ReversibleKeyGenerator
+impl<H: ReversibleStorageHasher, K: FullCodec> ReversibleKeyGenerator
 	for Key<H, K>
 {
 	type ReversibleHasher = H;
diff --git a/substrate/frame/support/src/storage/types/map.rs b/substrate/frame/support/src/storage/types/map.rs
index 3582448..12e992b 100644
--- a/substrate/frame/support/src/storage/types/map.rs
+++ b/substrate/frame/support/src/storage/types/map.rs
@@ -484,8 +484,8 @@ impl<Prefix, Hasher, Key, Value, QueryKind, OnEmpty, MaxValues> StorageEntryMeta
 where
 	Prefix: StorageInstance,
 	Hasher: crate::hash::StorageHasher,
-	Key: FullCodec + scale_info::StaticTypeInfo,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Key: FullCodec,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
@@ -502,8 +502,8 @@ where
 			modifier: QueryKind::METADATA,
 			ty: StorageEntryTypeIR::Map {
 				hashers: vec![Hasher::METADATA],
-				key: scale_info::meta_type::<Key>(),
-				value: scale_info::meta_type::<Value>(),
+				key: scale_info::meta_type::<()>(),
+				value: scale_info::meta_type::<()>(),
 			},
 			default: OnEmpty::get().encode(),
 			docs,
diff --git a/substrate/frame/support/src/storage/types/nmap.rs b/substrate/frame/support/src/storage/types/nmap.rs
index 0fc22b3..1b78f4e 100755
--- a/substrate/frame/support/src/storage/types/nmap.rs
+++ b/substrate/frame/support/src/storage/types/nmap.rs
@@ -592,7 +592,7 @@ impl<Prefix, Key, Value, QueryKind, OnEmpty, MaxValues> StorageEntryMetadataBuil
 where
 	Prefix: StorageInstance,
 	Key: super::key::KeyGenerator,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: Get<QueryKind::Query> + 'static,
 	MaxValues: Get<Option<u32>>,
@@ -608,9 +608,9 @@ where
 			name: Prefix::STORAGE_PREFIX,
 			modifier: QueryKind::METADATA,
 			ty: StorageEntryTypeIR::Map {
-				key: scale_info::meta_type::<Key::Key>(),
+				key: scale_info::meta_type::<()>(),
 				hashers: Key::HASHER_METADATA.to_vec(),
-				value: scale_info::meta_type::<Value>(),
+				value: scale_info::meta_type::<()>(),
 			},
 			default: OnEmpty::get().encode(),
 			docs,
diff --git a/substrate/frame/support/src/storage/types/value.rs b/substrate/frame/support/src/storage/types/value.rs
index 69a8f3b..40aaef3 100644
--- a/substrate/frame/support/src/storage/types/value.rs
+++ b/substrate/frame/support/src/storage/types/value.rs
@@ -273,7 +273,7 @@ impl<Prefix, Value, QueryKind, OnEmpty> StorageEntryMetadataBuilder
 	for StorageValue<Prefix, Value, QueryKind, OnEmpty>
 where
 	Prefix: StorageInstance,
-	Value: FullCodec + scale_info::StaticTypeInfo,
+	Value: FullCodec,
 	QueryKind: QueryKindTrait<Value, OnEmpty>,
 	OnEmpty: crate::traits::Get<QueryKind::Query> + 'static,
 {
@@ -287,7 +287,7 @@ where
 		let entry = StorageEntryMetadataIR {
 			name: Prefix::STORAGE_PREFIX,
 			modifier: QueryKind::METADATA,
-			ty: StorageEntryTypeIR::Plain(scale_info::meta_type::<Value>()),
+			ty: StorageEntryTypeIR::Plain(scale_info::meta_type::<()>()),
 			default: OnEmpty::get().encode(),
 			docs,
 			deprecation_info: deprecation_status,
