diff --git a/substrate/frame/support/procedural/src/construct_runtime/expand/call.rs b/substrate/frame/support/procedural/src/construct_runtime/expand/call.rs
index 4364811..da29e07 100644
--- a/substrate/frame/support/procedural/src/construct_runtime/expand/call.rs
+++ b/substrate/frame/support/procedural/src/construct_runtime/expand/call.rs
@@ -63,7 +63,6 @@ pub fn expand_outer_dispatch(
 			#scrate::__private::codec::Encode,
 			#scrate::__private::codec::Decode,
 			#scrate::__private::codec::DecodeWithMemTracking,
-			#scrate::__private::scale_info::TypeInfo,
 			#scrate::__private::RuntimeDebug,
 		)]
 		pub enum RuntimeCall {
diff --git a/substrate/frame/support/procedural/src/pallet/expand/call.rs b/substrate/frame/support/procedural/src/pallet/expand/call.rs
index a968b5f..96e68f0 100644
--- a/substrate/frame/support/procedural/src/pallet/expand/call.rs
+++ b/substrate/frame/support/procedural/src/pallet/expand/call.rs
@@ -307,11 +307,9 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 			#frame_support::__private::codec::Encode,
 			#frame_support::__private::codec::Decode,
 			#frame_support::__private::codec::DecodeWithMemTracking,
-			#frame_support::__private::scale_info::TypeInfo,
 		)]
 		#[codec(encode_bound())]
 		#[codec(decode_bound())]
-		#[scale_info(skip_type_params(#type_use_gen), capture_docs = #capture_docs)]
 		#[allow(non_camel_case_types)]
 		pub enum #call_ident<#type_decl_bounded_gen> #where_clause {
 			#[doc(hidden)]
@@ -476,7 +474,7 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 			#[doc(hidden)]
 			pub fn call_functions() -> #frame_support::__private::metadata_ir::PalletCallMetadataIR {
 				#frame_support::__private::metadata_ir::PalletCallMetadataIR  {
-					ty: #frame_support::__private::scale_info::meta_type::<#call_ident<#type_use_gen>>(),
+					ty: #frame_support::__private::scale_info::meta_type::<()>(),
 					deprecation_info: #deprecation,
 				}
 			}
diff --git a/substrate/frame/system/src/lib.rs b/substrate/frame/system/src/lib.rs
index 7c58874..460bc98 100644
--- a/substrate/frame/system/src/lib.rs
+++ b/substrate/frame/system/src/lib.rs
@@ -530,7 +530,7 @@ pub mod pallet {
 		#[docify::export(system_runtime_call)]
 		/// The aggregated `RuntimeCall` type.
 		#[pallet::no_default_bounds]
-		type RuntimeCall: Parameter
+		type RuntimeCall: codec::Codec + DecodeWithMemTracking + EncodeLike + Clone + Eq + core::fmt::Debug
 			+ Dispatchable<RuntimeOrigin = Self::RuntimeOrigin>
 			+ Debug
 			+ GetDispatchInfo
diff --git a/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs b/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs
index 6744b96..faf8a42 100644
--- a/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs
+++ b/substrate/primitives/runtime/src/generic/unchecked_extrinsic.rs
@@ -242,7 +242,7 @@ impl<Address, Call, Signature, Extension> TypeInfo
 	for UncheckedExtrinsic<Address, Call, Signature, Extension>
 where
 	Address: StaticTypeInfo,
-	Call: StaticTypeInfo,
+	Call: 'static,
 	Signature: StaticTypeInfo,
 	Extension: StaticTypeInfo,
 {
@@ -256,7 +256,7 @@ where
 			// to help construct the custom decoding from the opaque bytes (see below).
 			.type_params(vec![
 				TypeParameter::new("Address", Some(meta_type::<Address>())),
-				TypeParameter::new("Call", Some(meta_type::<Call>())),
+				TypeParameter::new("Call", Some(meta_type::<()>())),
 				TypeParameter::new("Signature", Some(meta_type::<Signature>())),
 				TypeParameter::new("Extra", Some(meta_type::<Extension>())),
 			])
@@ -319,7 +319,7 @@ impl<Address, Call, Signature, Extension> UncheckedExtrinsic<Address, Call, Sign
 	}
 }
 
-impl<Address: TypeInfo, Call: TypeInfo, Signature: TypeInfo, Extension: TypeInfo> ExtrinsicLike
+impl<Address: TypeInfo, Call, Signature: TypeInfo, Extension: TypeInfo> ExtrinsicLike
 	for UncheckedExtrinsic<Address, Call, Signature, Extension>
 {
 	fn is_bare(&self) -> bool {
