diff --git a/polkadot-sdk/substrate/frame/system/src/lib.rs b/polkadot-sdk/substrate/frame/system/src/lib.rs
index be3c1f4..afb236a 100644
--- a/polkadot-sdk/substrate/frame/system/src/lib.rs
+++ b/polkadot-sdk/substrate/frame/system/src/lib.rs
@@ -1099,8 +1099,8 @@ pub mod pallet {
 	#[pallet::genesis_build]
 	impl<T: Config> BuildGenesisConfig for GenesisConfig<T> {
 		fn build(&self) {
-			<BlockHash<T>>::insert::<_, T::Hash>(BlockNumberFor::<T>::zero(), hash69());
-			<ParentHash<T>>::put::<T::Hash>(hash69());
+			<BlockHash<T>>::insert::<_, T::Hash>(BlockNumberFor::<T>::zero(), hash0());
+			<ParentHash<T>>::put::<T::Hash>(hash0());
 			<LastRuntimeUpgrade<T>>::put(LastRuntimeUpgradeInfo::from(T::Version::get()));
 			<UpgradedToU32RefCount<T>>::put(true);
 			<UpgradedToTripleRefCount<T>>::put(true);
@@ -1178,11 +1178,11 @@ pub struct EventRecord<E: Parameter + Member, T> {
 	pub topics: Vec<T>,
 }
 
-// Create a Hash with 69 for each byte,
+// Create a Hash with 0 for each byte,
 // only used to build genesis config.
-fn hash69<T: AsMut<[u8]> + Default>() -> T {
+fn hash0<T: AsMut<[u8]> + Default>() -> T {
 	let mut h = T::default();
-	h.as_mut().iter_mut().for_each(|byte| *byte = 69);
+	h.as_mut().iter_mut().for_each(|byte| *byte = 0);
 	h
 }
 
