diff --git a/substrate/frame/src/lib.rs b/substrate/frame/src/lib.rs
index 72ca417..e0d5d7e 100644
--- a/substrate/frame/src/lib.rs
+++ b/substrate/frame/src/lib.rs
@@ -445,7 +445,6 @@ pub mod runtime {
 		pub use frame_system_rpc_runtime_api::*;
 		pub use sp_api::{self, *};
 		pub use sp_block_builder::*;
-		pub use sp_consensus_aura::*;
 		pub use sp_consensus_grandpa::*;
 		pub use sp_genesis_builder::*;
 		pub use sp_offchain::*;
@@ -642,8 +641,6 @@ pub mod deps {
 	#[cfg(feature = "runtime")]
 	pub use sp_block_builder;
 	#[cfg(feature = "runtime")]
-	pub use sp_consensus_aura;
-	#[cfg(feature = "runtime")]
 	pub use sp_consensus_grandpa;
 	#[cfg(feature = "runtime")]
 	pub use sp_genesis_builder;
