diff --git a/substrate/utils/wasm-builder/src/lib.rs b/substrate/utils/wasm-builder/src/lib.rs
index c28f3c3..1bb0e45 100644
--- a/substrate/utils/wasm-builder/src/lib.rs
+++ b/substrate/utils/wasm-builder/src/lib.rs
@@ -407,8 +407,8 @@ fn get_bool_environment_variable(name: &str) -> Option<bool> {
 	} else if value == "0" {
 		Some(false)
 	} else {
-		build_helper::warning!(
-			"the '{name}' environment variable has an invalid value; it must be either '1' or '0'",
+		println!(
+			"cargo:warning=the '{name}' environment variable has an invalid value; it must be either '1' or '0'",
 		);
 		std::process::exit(1);
 	}
@@ -432,8 +432,8 @@ impl RuntimeTarget {
 		} else if value == "riscv" {
 			Self::Riscv
 		} else {
-			build_helper::warning!(
-				"RUNTIME_TARGET environment variable must be set to either \"wasm\" or \"riscv\""
+			println!(
+				"cargo:warning=RUNTIME_TARGET environment variable must be set to either \"wasm\" or \"riscv\""
 			);
 			std::process::exit(1);
 		}
diff --git a/substrate/utils/wasm-builder/src/prerequisites.rs b/substrate/utils/wasm-builder/src/prerequisites.rs
index 727197f..0cbd10a 100644
--- a/substrate/utils/wasm-builder/src/prerequisites.rs
+++ b/substrate/utils/wasm-builder/src/prerequisites.rs
@@ -270,9 +270,9 @@ fn check_wasm_toolchain_installed(
 	if cargo_command.supports_wasm32v1_none_target() &&
 		!cargo_command.is_wasm32v1_none_target_installed()
 	{
-		build_helper::warning!("You are building WASM runtime using `wasm32-unknown-unknown` target, although Rust >= 1.84 supports `wasm32v1-none` target!");
-		build_helper::warning!("You can install it with `rustup target add wasm32v1-none --toolchain {toolchain}` if you're using `rustup`.");
-		build_helper::warning!("After installing `wasm32v1-none` target, you must rebuild WASM runtime from scratch, use `cargo clean` before building.");
+		println!("cargo:warning=You are building WASM runtime using `wasm32-unknown-unknown` target, although Rust >= 1.84 supports `wasm32v1-none` target!");
+		println!("cargo:warning=You can install it with `rustup target add wasm32v1-none --toolchain {toolchain}` if you're using `rustup`.");
+		println!("cargo:warning=After installing `wasm32v1-none` target, you must rebuild WASM runtime from scratch, use `cargo clean` before building.");
 	}
 
 	Ok(CargoCommandVersioned::new(cargo_command, version))
diff --git a/substrate/utils/wasm-builder/src/wasm_project.rs b/substrate/utils/wasm-builder/src/wasm_project.rs
index 19f9692..1b47a77 100644
--- a/substrate/utils/wasm-builder/src/wasm_project.rs
+++ b/substrate/utils/wasm-builder/src/wasm_project.rs
@@ -19,7 +19,6 @@
 use crate::builder::MetadataExtraInfo;
 use crate::{write_file_if_changed, CargoCommandVersioned, RuntimeTarget, OFFLINE};
 
-use build_helper::rerun_if_changed;
 use cargo_metadata::{DependencyKind, Metadata, MetadataCommand};
 use console::style;
 use parity_wasm::elements::{deserialize_buffer, Module};
@@ -229,7 +228,7 @@ pub(crate) fn create_and_compile(
 	);
 
 	if let Err(err) = adjust_mtime(&bloaty_blob_binary, final_blob_binary.as_ref()) {
-		build_helper::warning!("Error while adjusting the mtime of the blob binaries: {}", err)
+		println!("cargo:warning=Error while adjusting the mtime of the blob binaries: {}", err)
 	}
 
 	(final_blob_binary, bloaty_blob_binary)
@@ -313,7 +312,7 @@ fn adjust_mtime(
 	bloaty_wasm: &WasmBinaryBloaty,
 	compressed_or_compact_wasm: Option<&WasmBinary>,
 ) -> std::io::Result<()> {
-	let out_dir = build_helper::out_dir();
+	let out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
 	let invoked_timestamp = out_dir.join("../invoked.timestamp");
 
 	// Get the mtime of the `invoked.timestamp`
@@ -350,22 +349,22 @@ fn find_cargo_lock(cargo_manifest: &Path) -> Option<PathBuf> {
 		if path.join("Cargo.lock").exists() {
 			return Some(path.join("Cargo.lock"))
 		} else {
-			build_helper::warning!(
-				"`{}` env variable doesn't point to a directory that contains a `Cargo.lock`.",
+			println!(
+				"cargo:warning=`{}` env variable doesn't point to a directory that contains a `Cargo.lock`.",
 				crate::WASM_BUILD_WORKSPACE_HINT,
 			);
 		}
 	}
 
-	if let Some(path) = find_impl(build_helper::out_dir()) {
+	if let Some(path) = find_impl(PathBuf::from(std::env::var("OUT_DIR").unwrap())) {
 		return Some(path)
 	}
 
-	build_helper::warning!(
-		"Could not find `Cargo.lock` for `{}`, while searching from `{}`. \
+	println!(
+		"cargo:warning=Could not find `Cargo.lock` for `{}`, while searching from `{}`. \
 		 To fix this, point the `{}` env variable to the directory of the workspace being compiled.",
 		cargo_manifest.display(),
-		build_helper::out_dir().display(),
+		PathBuf::from(std::env::var("OUT_DIR").unwrap()).display(),
 		crate::WASM_BUILD_WORKSPACE_HINT,
 	);
 
@@ -415,7 +414,7 @@ fn get_blob_name(target: RuntimeTarget, cargo_manifest: &Path) -> String {
 
 /// Returns the root path of the wasm workspace.
 fn get_wasm_workspace_root() -> PathBuf {
-	let mut out_dir = build_helper::out_dir();
+	let mut out_dir = PathBuf::from(std::env::var("OUT_DIR").unwrap());
 
 	loop {
 		match out_dir.parent() {
@@ -427,7 +426,7 @@ fn get_wasm_workspace_root() -> PathBuf {
 		}
 	}
 
-	panic!("Could not find target dir in: {}", build_helper::out_dir().display())
+	panic!("Could not find target dir in: {}", PathBuf::from(std::env::var("OUT_DIR").unwrap()).display())
 }
 
 fn create_project_cargo_toml(
@@ -802,8 +801,8 @@ impl BuildConfiguration {
 			// not doing a fallback.
 			(None, false) => {
 				let profile = Profile::Release;
-				build_helper::warning!(
-					"Unknown cargo profile `{name}`. Defaulted to `{profile:?}` for the runtime build.",
+				println!(
+					"cargo:warning=Unknown cargo profile `{name}`. Defaulted to `{profile:?}` for the runtime build.",
 				);
 				profile
 			},
@@ -893,10 +892,10 @@ fn build_bloaty_blob(
 	let cargo_args = env::var(crate::WASM_BUILD_CARGO_ARGS).unwrap_or_default();
 	if !cargo_args.is_empty() {
 		let Some(args) = shlex::split(&cargo_args) else {
-			build_helper::warning(format!(
-				"the {} environment variable is not a valid shell string",
+			println!(
+				"cargo:warning=the {} environment variable is not a valid shell string",
 				crate::WASM_BUILD_CARGO_ARGS
-			));
+			);
 			std::process::exit(1);
 		};
 		build_cmd.args(args);
@@ -1061,8 +1060,8 @@ fn try_compress_blob(compact_blob_path: &Path, out_name: &str) -> Option<WasmBin
 		);
 		Some(WasmBinary(compact_compressed_blob_path))
 	} else {
-		build_helper::warning!(
-			"Writing uncompressed blob. Exceeded maximum size {}",
+		println!(
+			"cargo:warning=Writing uncompressed blob. Exceeded maximum size {}",
 			CODE_BLOB_BOMB_LIMIT,
 		);
 		println!("{}", colorize_info_message("Skipping blob compression"));
@@ -1130,7 +1129,7 @@ fn generate_rerun_if_changed_instructions(
 ) {
 	// Rerun `build.rs` if the `Cargo.lock` changes
 	if let Some(cargo_lock) = find_cargo_lock(cargo_manifest) {
-		rerun_if_changed(cargo_lock);
+		println!("cargo:rerun-if-changed={}", cargo_lock.display());
 	}
 
 	let metadata = create_metadata_command(project_folder.join("Cargo.toml"))
@@ -1180,8 +1179,8 @@ fn generate_rerun_if_changed_instructions(
 	// Make sure that if any file/folder of a dependency change, we need to rerun the `build.rs`
 	packages.iter().for_each(package_rerun_if_changed);
 
-	compressed_or_compact_wasm.map(|w| rerun_if_changed(w.wasm_binary_path()));
-	rerun_if_changed(bloaty_wasm.bloaty_path());
+	compressed_or_compact_wasm.map(|w| println!("cargo:rerun-if-changed={}", w.wasm_binary_path().display()));
+	println!("cargo:rerun-if-changed={}", bloaty_wasm.bloaty_path().display());
 
 	// Register our env variables
 	println!("cargo:rerun-if-env-changed={}", crate::SKIP_BUILD_ENV);
@@ -1212,7 +1211,7 @@ fn package_rerun_if_changed(package: &DeduplicatePackage) {
 		})
 		.filter_map(|p| p.ok().map(|p| p.into_path()))
 		.filter(|p| p.extension().map(|e| e == "rs" || e == "toml").unwrap_or_default())
-		.for_each(rerun_if_changed);
+		.for_each(|i| println!("cargo:rerun-if-changed={}", i.display()));
 }
 
 /// Copy the blob binary to the target directory set in `WASM_TARGET_DIRECTORY` environment
