diff --git a/substrate/frame/support/procedural/src/construct_runtime/mod.rs b/substrate/frame/support/procedural/src/construct_runtime/mod.rs
index 8e32697..964a619 100644
--- a/substrate/frame/support/procedural/src/construct_runtime/mod.rs
+++ b/substrate/frame/support/procedural/src/construct_runtime/mod.rs
@@ -399,7 +399,7 @@ fn construct_runtime_final_expansion(
 	let pallet_to_index = decl_pallet_runtime_setup(&name, &pallets, &scrate);
 
 	let dispatch = expand::expand_outer_dispatch(&name, system_pallet, &pallets, &scrate);
-	let tasks = expand::expand_outer_task(&name, &pallets, &scrate);
+	// let tasks = expand::expand_outer_task(&name, &pallets, &scrate);
 	let query = expand::expand_outer_query(&name, &pallets, &scrate);
 	let metadata = expand::expand_runtime_metadata(
 		&name,
@@ -412,10 +412,10 @@ fn construct_runtime_final_expansion(
 	let inherent =
 		expand::expand_outer_inherent(&name, &block, &unchecked_extrinsic, &pallets, &scrate);
 	let validate_unsigned = expand::expand_outer_validate_unsigned(&name, &pallets, &scrate);
-	let freeze_reason = expand::expand_outer_freeze_reason(&pallets, &scrate);
+	/* let freeze_reason = expand::expand_outer_freeze_reason(&pallets, &scrate);
 	let hold_reason = expand::expand_outer_hold_reason(&pallets, &scrate);
 	let lock_id = expand::expand_outer_lock_id(&pallets, &scrate);
-	let slash_reason = expand::expand_outer_slash_reason(&pallets, &scrate);
+	let slash_reason = expand::expand_outer_slash_reason(&pallets, &scrate); */
 	let integrity_test = decl_integrity_test(&scrate);
 	let static_assertions = decl_static_assertions(&name, &pallets, &scrate);
 
@@ -491,7 +491,7 @@ fn construct_runtime_final_expansion(
 
 		#dispatch
 
-		#tasks
+		// #tasks
 
 		#query
 
@@ -503,13 +503,13 @@ fn construct_runtime_final_expansion(
 
 		#validate_unsigned
 
-		#freeze_reason
+		/* #freeze_reason
 
 		#hold_reason
 
 		#lock_id
 
-		#slash_reason
+		#slash_reason */
 
 		#integrity_test
 
diff --git a/substrate/frame/support/procedural/src/construct_runtime/parse.rs b/substrate/frame/support/procedural/src/construct_runtime/parse.rs
index 2df0812..2b15ee8 100644
--- a/substrate/frame/support/procedural/src/construct_runtime/parse.rs
+++ b/substrate/frame/support/procedural/src/construct_runtime/parse.rs
@@ -41,11 +41,11 @@ mod keyword {
 	syn::custom_keyword!(Origin);
 	syn::custom_keyword!(Inherent);
 	syn::custom_keyword!(ValidateUnsigned);
-	syn::custom_keyword!(FreezeReason);
+	/* syn::custom_keyword!(FreezeReason);
 	syn::custom_keyword!(HoldReason);
 	syn::custom_keyword!(Task);
 	syn::custom_keyword!(LockId);
-	syn::custom_keyword!(SlashReason);
+	syn::custom_keyword!(SlashReason); */
 	syn::custom_keyword!(exclude_parts);
 	syn::custom_keyword!(use_parts);
 	syn::custom_keyword!(expanded);
@@ -389,11 +389,11 @@ pub enum PalletPartKeyword {
 	Origin(keyword::Origin),
 	Inherent(keyword::Inherent),
 	ValidateUnsigned(keyword::ValidateUnsigned),
-	FreezeReason(keyword::FreezeReason),
+	/* FreezeReason(keyword::FreezeReason),
 	HoldReason(keyword::HoldReason),
 	Task(keyword::Task),
 	LockId(keyword::LockId),
-	SlashReason(keyword::SlashReason),
+	SlashReason(keyword::SlashReason), */
 }
 
 impl Parse for PalletPartKeyword {
@@ -418,7 +418,7 @@ impl Parse for PalletPartKeyword {
 			Ok(Self::Inherent(input.parse()?))
 		} else if lookahead.peek(keyword::ValidateUnsigned) {
 			Ok(Self::ValidateUnsigned(input.parse()?))
-		} else if lookahead.peek(keyword::FreezeReason) {
+		/* } else if lookahead.peek(keyword::FreezeReason) {
 			Ok(Self::FreezeReason(input.parse()?))
 		} else if lookahead.peek(keyword::HoldReason) {
 			Ok(Self::HoldReason(input.parse()?))
@@ -427,7 +427,7 @@ impl Parse for PalletPartKeyword {
 		} else if lookahead.peek(keyword::LockId) {
 			Ok(Self::LockId(input.parse()?))
 		} else if lookahead.peek(keyword::SlashReason) {
-			Ok(Self::SlashReason(input.parse()?))
+			Ok(Self::SlashReason(input.parse()?)) */
 		} else {
 			Err(lookahead.error())
 		}
@@ -447,11 +447,11 @@ impl PalletPartKeyword {
 			Self::Origin(_) => "Origin",
 			Self::Inherent(_) => "Inherent",
 			Self::ValidateUnsigned(_) => "ValidateUnsigned",
-			Self::FreezeReason(_) => "FreezeReason",
+			/* Self::FreezeReason(_) => "FreezeReason",
 			Self::HoldReason(_) => "HoldReason",
 			Self::Task(_) => "Task",
 			Self::LockId(_) => "LockId",
-			Self::SlashReason(_) => "SlashReason",
+			Self::SlashReason(_) => "SlashReason", */
 		}
 	}
 
@@ -462,7 +462,7 @@ impl PalletPartKeyword {
 
 	/// Returns the names of all pallet parts that allow to have a generic argument.
 	fn all_generic_arg() -> &'static [&'static str] {
-		&["Event", "Error", "Origin", "Config", "Task"]
+		&["Event", "Error", "Origin", "Config"]
 	}
 }
 
@@ -478,11 +478,11 @@ impl ToTokens for PalletPartKeyword {
 			Self::Origin(inner) => inner.to_tokens(tokens),
 			Self::Inherent(inner) => inner.to_tokens(tokens),
 			Self::ValidateUnsigned(inner) => inner.to_tokens(tokens),
-			Self::FreezeReason(inner) => inner.to_tokens(tokens),
+			/* Self::FreezeReason(inner) => inner.to_tokens(tokens),
 			Self::HoldReason(inner) => inner.to_tokens(tokens),
 			Self::Task(inner) => inner.to_tokens(tokens),
 			Self::LockId(inner) => inner.to_tokens(tokens),
-			Self::SlashReason(inner) => inner.to_tokens(tokens),
+			Self::SlashReason(inner) => inner.to_tokens(tokens), */
 		}
 	}
 }
diff --git a/substrate/frame/support/procedural/src/lib.rs b/substrate/frame/support/procedural/src/lib.rs
index 4c24e18..76b1ca1 100644
--- a/substrate/frame/support/procedural/src/lib.rs
+++ b/substrate/frame/support/procedural/src/lib.rs
@@ -817,11 +817,11 @@ pub fn inject_runtime_type(_: TokenStream, tokens: TokenStream) -> TokenStream {
 	let item = syn::parse_macro_input!(item as TraitItemType);
 	if item.ident != "RuntimeCall" &&
 		item.ident != "RuntimeEvent" &&
-		item.ident != "RuntimeTask" &&
+		// item.ident != "RuntimeTask" &&
 		item.ident != "RuntimeViewFunction" &&
 		item.ident != "RuntimeOrigin" &&
-		item.ident != "RuntimeHoldReason" &&
-		item.ident != "RuntimeFreezeReason" &&
+		// item.ident != "RuntimeHoldReason" &&
+		// item.ident != "RuntimeFreezeReason" &&
 		item.ident != "RuntimeParameters" &&
 		item.ident != "PalletInfo"
 	{
@@ -1035,6 +1035,6 @@ pub fn origin(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
 }
 
-/// ---
+/* /// ---
 ///
 /// Documentation for this macro can be found at `frame_support::pallet_macros::composite_enum`.
@@ -1204,6 +1204,6 @@ pub fn task_weight(_: TokenStream, _: TokenStream) -> TokenStream {
 #[proc_macro_attribute]
 pub fn task_index(_: TokenStream, _: TokenStream) -> TokenStream {
 	pallet_macro_stub()
-}
+} */

 /// ---
diff --git a/substrate/frame/support/procedural/src/pallet/expand/mod.rs b/substrate/frame/support/procedural/src/pallet/expand/mod.rs
index c8bf5ca..94c0611 100644
--- a/substrate/frame/support/procedural/src/pallet/expand/mod.rs
+++ b/substrate/frame/support/procedural/src/pallet/expand/mod.rs
@@ -63,7 +63,7 @@ pub fn expand(mut def: Def) -> proc_macro2::TokenStream {
 	let config = config::expand_config(&mut def);
 	let associated_types = config::expand_config_metadata(&def);
 	let call = call::expand_call(&mut def);
-	let tasks = tasks::expand_tasks(&mut def);
+	// let tasks = tasks::expand_tasks(&mut def);
 	let error = error::expand_error(&mut def);
 	let event = event::expand_event(&mut def);
 	let storages = storage::expand_storages(&mut def);
@@ -78,7 +78,7 @@ pub fn expand(mut def: Def) -> proc_macro2::TokenStream {
 	let validate_unsigned = validate_unsigned::expand_validate_unsigned(&mut def);
 	let tt_default_parts = tt_default_parts::expand_tt_default_parts(&mut def);
 	let doc_only = doc_only::expand_doc_only(&mut def);
-	let composites = composite::expand_composites(&mut def);
+	// let composites = composite::expand_composites(&mut def);
 
 	let warnings = def.config.warnings;
 
@@ -111,7 +111,7 @@ storage item. Otherwise, all storage items are listed among [*Type Definitions*]
 		#config
 		#associated_types
 		#call
-		#tasks
+		// #tasks
 		#error
 		#event
 		#storages
@@ -126,7 +126,7 @@ storage item. Otherwise, all storage items are listed among [*Type Definitions*]
 		#validate_unsigned
 		#tt_default_parts
 		#doc_only
-		#composites
+		// #composites
 	);
 
 	let item = &mut def.item.content.as_mut().expect("This is checked by parsing").1;
diff --git a/substrate/frame/support/procedural/src/pallet/expand/tt_default_parts.rs b/substrate/frame/support/procedural/src/pallet/expand/tt_default_parts.rs
index 6d53de3..d201c65 100644
--- a/substrate/frame/support/procedural/src/pallet/expand/tt_default_parts.rs
+++ b/substrate/frame/support/procedural/src/pallet/expand/tt_default_parts.rs
@@ -16,7 +16,7 @@
 // limitations under the License.
 
 use crate::{
-	pallet::{CompositeKeyword, Def},
+	pallet::Def,
 	COUNTER,
 };
 use syn::spanned::Spanned;
@@ -33,7 +33,7 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 
 	let call_part = def.call.as_ref().map(|_| quote::quote!(Call,));
 
-	let task_part = def.tasks.as_ref().map(|_| quote::quote!(Task,));
+	// let task_part = def.tasks.as_ref().map(|_| quote::quote!(Task,));
 
 	let storage_part = (!def.storages.is_empty()).then(|| quote::quote!(Storage,));
 
@@ -59,7 +59,7 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 	let validate_unsigned_part =
 		def.validate_unsigned.as_ref().map(|_| quote::quote!(ValidateUnsigned,));
 
-	let freeze_reason_part = def
+	/* let freeze_reason_part = def
 		.composites
 		.iter()
 		.any(|c| matches!(c.composite_keyword, CompositeKeyword::FreezeReason(_)))
@@ -81,11 +81,11 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 		.composites
 		.iter()
 		.any(|c| matches!(c.composite_keyword, CompositeKeyword::SlashReason(_)))
-		.then_some(quote::quote!(SlashReason,));
+		.then_some(quote::quote!(SlashReason,)); */
 
 	let call_part_v2 = def.call.as_ref().map(|_| quote::quote!(+ Call));
 
-	let task_part_v2 = def.tasks.as_ref().map(|_| quote::quote!(+ Task));
+	// let task_part_v2 = def.tasks.as_ref().map(|_| quote::quote!(+ Task));
 
 	let storage_part_v2 = (!def.storages.is_empty()).then(|| quote::quote!(+ Storage));
 
@@ -111,7 +111,7 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 	let validate_unsigned_part_v2 =
 		def.validate_unsigned.as_ref().map(|_| quote::quote!(+ ValidateUnsigned));
 
-	let freeze_reason_part_v2 = def
+	/* let freeze_reason_part_v2 = def
 		.composites
 		.iter()
 		.any(|c| matches!(c.composite_keyword, CompositeKeyword::FreezeReason(_)))
@@ -133,7 +133,7 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 		.composites
 		.iter()
 		.any(|c| matches!(c.composite_keyword, CompositeKeyword::SlashReason(_)))
-		.then_some(quote::quote!(+ SlashReason));
+		.then_some(quote::quote!(+ SlashReason)); */
 
 	quote::quote!(
 		// This macro follows the conventions as laid out by the `tt-call` crate. It does not
@@ -155,8 +155,8 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 					tokens = [{
 						expanded::{
 							Pallet, #call_part #storage_part #event_part #error_part #origin_part #config_part
-							#inherent_part #validate_unsigned_part #freeze_reason_part #task_part
-							#hold_reason_part #lock_id_part #slash_reason_part
+							#inherent_part #validate_unsigned_part /* #freeze_reason_part #task_part
+							#hold_reason_part #lock_id_part #slash_reason_part */
 						}
 					}]
 				}
@@ -204,8 +204,8 @@ pub fn expand_tt_default_parts(def: &mut Def) -> proc_macro2::TokenStream {
 					$caller
 					tokens = [{
 						+ Pallet #call_part_v2 #storage_part_v2 #event_part_v2 #error_part_v2 #origin_part_v2 #config_part_v2
-						#inherent_part_v2 #validate_unsigned_part_v2 #freeze_reason_part_v2 #task_part_v2
-						#hold_reason_part_v2 #lock_id_part_v2 #slash_reason_part_v2
+						#inherent_part_v2 #validate_unsigned_part_v2 /* #freeze_reason_part_v2 #task_part_v2
+						#hold_reason_part_v2 #lock_id_part_v2 #slash_reason_part_v2 */
 					}]
 				}
 			};
diff --git a/substrate/frame/support/procedural/src/pallet/mod.rs b/substrate/frame/support/procedural/src/pallet/mod.rs
index 42d8272..d9897ea 100644
--- a/substrate/frame/support/procedural/src/pallet/mod.rs
+++ b/substrate/frame/support/procedural/src/pallet/mod.rs
@@ -28,7 +28,7 @@
 mod expand;
 pub(crate) mod parse;
 
-pub use parse::{composite::keyword::CompositeKeyword, Def};
+pub use parse::Def;
 use syn::spanned::Spanned;
 
 mod keyword {
diff --git a/substrate/frame/support/procedural/src/pallet/parse/mod.rs b/substrate/frame/support/procedural/src/pallet/parse/mod.rs
index 8d94c49..c92e8c0 100644
--- a/substrate/frame/support/procedural/src/pallet/parse/mod.rs
+++ b/substrate/frame/support/procedural/src/pallet/parse/mod.rs
@@ -55,8 +55,8 @@ pub struct Def {
 	pub pallet_struct: pallet_struct::PalletStructDef,
 	pub hooks: Option<hooks::HooksDef>,
 	pub call: Option<call::CallDef>,
-	pub tasks: Option<tasks::TasksDef>,
-	pub task_enum: Option<tasks::TaskEnumDef>,
+	/* pub tasks: Option<tasks::TasksDef>,
+	pub task_enum: Option<tasks::TaskEnumDef>, */
 	pub storages: Vec<storage::StorageDef>,
 	pub error: Option<error::ErrorDef>,
 	pub event: Option<event::EventDef>,
@@ -66,7 +66,7 @@ pub struct Def {
 	pub genesis_build: Option<genesis_build::GenesisBuildDef>,
 	pub validate_unsigned: Option<validate_unsigned::ValidateUnsignedDef>,
 	pub extra_constants: Option<extra_constants::ExtraConstantsDef>,
-	pub composites: Vec<composite::CompositeDef>,
+	// pub composites: Vec<composite::CompositeDef>,
 	pub type_values: Vec<type_value::TypeValueDef>,
 	pub frame_system: syn::Path,
 	pub frame_support: syn::Path,
@@ -93,8 +93,8 @@ impl Def {
 		let mut pallet_struct = None;
 		let mut hooks = None;
 		let mut call = None;
-		let mut tasks = None;
-		let mut task_enum = None;
+		/* let mut tasks = None;
+		let mut task_enum = None; */
 		let mut error = None;
 		let mut event = None;
 		let mut origin = None;
@@ -105,7 +105,7 @@ impl Def {
 		let mut extra_constants = None;
 		let mut storages = vec![];
 		let mut type_values = vec![];
-		let mut composites: Vec<CompositeDef> = vec![];
+		// let mut composites: Vec<CompositeDef> = vec![];
 		let mut view_functions = None;
 		let mut is_frame_system = false;
 
@@ -134,7 +134,7 @@ impl Def {
 				},
 				Some(PalletAttr::RuntimeCall(cw, span)) if call.is_none() =>
 					call = Some(call::CallDef::try_from(span, index, item, dev_mode, cw)?),
-				Some(PalletAttr::Tasks(span)) if tasks.is_none() => {
+				/* Some(PalletAttr::Tasks(span)) if tasks.is_none() => {
 					let item_tokens = item.to_token_stream();
 					// `TasksDef::parse` needs to know if attr was provided so we artificially
 					// re-insert it here
@@ -159,7 +159,7 @@ impl Def {
 					"`#[pallet::task_list]` can only be used on items within an `impl` statement."
 				)),
 				Some(PalletAttr::RuntimeTask(_)) if task_enum.is_none() =>
-					task_enum = Some(syn::parse2::<tasks::TaskEnumDef>(item.to_token_stream())?),
+					task_enum = Some(syn::parse2::<tasks::TaskEnumDef>(item.to_token_stream())?), */
 				Some(PalletAttr::Error(span)) if error.is_none() =>
 					error = Some(error::ErrorDef::try_from(span, index, item)?),
 				Some(PalletAttr::RuntimeEvent(span)) if event.is_none() =>
@@ -187,7 +187,7 @@ impl Def {
 				Some(PalletAttr::ExtraConstants(_)) =>
 					extra_constants =
 						Some(extra_constants::ExtraConstantsDef::try_from(item)?),
-				Some(PalletAttr::Composite(span)) => {
+				/* Some(PalletAttr::Composite(span)) => {
 					let composite =
 						composite::CompositeDef::try_from(span, &frame_support, item)?;
 					if composites.iter().any(|def| {
@@ -212,7 +212,7 @@ impl Def {
 						return Err(syn::Error::new(composite.composite_keyword.span(), &msg))
 					}
 					composites.push(composite);
-				},
+				}, */
 				Some(PalletAttr::ViewFunctions(span)) => {
 					view_functions = Some(view_functions::ViewFunctionsImplDef::try_from(span, item)?);
 				}
@@ -235,7 +235,7 @@ impl Def {
 			return Err(syn::Error::new(item_span, msg));
 		}
 
-		Self::resolve_tasks(&item_span, &mut tasks, &mut task_enum, items)?;
+		// Self::resolve_tasks(&item_span, &mut tasks, &mut task_enum, items)?;
 
 		let def = Def {
 			item,
@@ -245,8 +245,8 @@ impl Def {
 				.ok_or_else(|| syn::Error::new(item_span, "Missing `#[pallet::pallet]`"))?,
 			hooks,
 			call,
-			tasks,
-			task_enum,
+			/* tasks,
+			task_enum, */
 			extra_constants,
 			genesis_config,
 			genesis_build,
@@ -256,7 +256,7 @@ impl Def {
 			origin,
 			inherent,
 			storages,
-			composites,
+			// composites,
 			type_values,
 			frame_system,
 			frame_support,
@@ -270,7 +270,7 @@ impl Def {
 		Ok(def)
 	}
 
-	/// Performs extra logic checks necessary for the `#[pallet::tasks_experimental]` feature.
+	/* /// Performs extra logic checks necessary for the `#[pallet::tasks_experimental]` feature.
 	fn resolve_tasks(
 		item_span: &proc_macro2::Span,
 		tasks: &mut Option<tasks::TasksDef>,
@@ -361,7 +361,7 @@ impl Def {
 		}
 		*tasks = result;
 		Ok(())
-	}
+	} */
 
 	/// Check that usage of trait `Config` is consistent with the definition, i.e. it is used with
 	/// instance iff it is defined with instance.
@@ -396,9 +396,9 @@ impl Def {
 		if let Some(extra_constants) = &self.extra_constants {
 			instances.extend_from_slice(&extra_constants.instances[..]);
 		}
-		if let Some(task_enum) = &self.task_enum {
+		/* if let Some(task_enum) = &self.task_enum {
 			instances.push(task_enum.instance_usage.clone());
-		}
+		} */
 
 		let mut errors = instances.into_iter().filter_map(|instances| {
 			if instances.has_instance == self.config.has_instance {
@@ -534,11 +534,11 @@ impl GenericKind {
 mod keyword {
 	syn::custom_keyword!(origin);
 	syn::custom_keyword!(call);
-	syn::custom_keyword!(tasks_experimental);
+	/* syn::custom_keyword!(tasks_experimental);
 	syn::custom_keyword!(task_enum);
 	syn::custom_keyword!(task_list);
 	syn::custom_keyword!(task_condition);
-	syn::custom_keyword!(task_index);
+	syn::custom_keyword!(task_index); */
 	syn::custom_keyword!(weight);
 	syn::custom_keyword!(event);
 	syn::custom_keyword!(config);
@@ -555,7 +555,7 @@ mod keyword {
 	syn::custom_keyword!(type_value);
 	syn::custom_keyword!(pallet);
 	syn::custom_keyword!(extra_constants);
-	syn::custom_keyword!(composite_enum);
+	// syn::custom_keyword!(composite_enum);
 	syn::custom_keyword!(view_functions);
 }
 
@@ -636,11 +636,11 @@ enum PalletAttr {
 	/// instead of the zero weight. So to say: it works together with `dev_mode`.
 	RuntimeCall(Option<InheritedCallWeightAttr>, proc_macro2::Span),
 	Error(proc_macro2::Span),
-	Tasks(proc_macro2::Span),
+	/* Tasks(proc_macro2::Span),
 	TaskList(proc_macro2::Span),
 	TaskCondition(proc_macro2::Span),
 	TaskIndex(proc_macro2::Span),
-	RuntimeTask(proc_macro2::Span),
+	RuntimeTask(proc_macro2::Span), */
 	RuntimeEvent(proc_macro2::Span),
 	RuntimeOrigin(proc_macro2::Span),
 	Inherent(proc_macro2::Span),
@@ -650,7 +650,7 @@ enum PalletAttr {
 	ValidateUnsigned(proc_macro2::Span),
 	TypeValue(proc_macro2::Span),
 	ExtraConstants(proc_macro2::Span),
-	Composite(proc_macro2::Span),
+	// Composite(proc_macro2::Span),
 	ViewFunctions(proc_macro2::Span),
 }
 
@@ -660,12 +660,12 @@ impl PalletAttr {
 			Self::Config { span, .. } => *span,
 			Self::Pallet(span) => *span,
 			Self::Hooks(span) => *span,
-			Self::Tasks(span) => *span,
+			/* Self::Tasks(span) => *span,
 			Self::TaskCondition(span) => *span,
 			Self::TaskIndex(span) => *span,
-			Self::TaskList(span) => *span,
+			Self::TaskList(span) => *span, */
 			Self::Error(span) => *span,
-			Self::RuntimeTask(span) => *span,
+			// Self::RuntimeTask(span) => *span,
 			Self::RuntimeCall(_, span) => *span,
 			Self::RuntimeEvent(span) => *span,
 			Self::RuntimeOrigin(span) => *span,
@@ -676,7 +676,7 @@ impl PalletAttr {
 			Self::ValidateUnsigned(span) => *span,
 			Self::TypeValue(span) => *span,
 			Self::ExtraConstants(span) => *span,
-			Self::Composite(span) => *span,
+			// Self::Composite(span) => *span,
 			Self::ViewFunctions(span) => *span,
 		}
 	}
@@ -763,7 +763,7 @@ impl syn::parse::Parse for PalletAttr {
 				false => Some(InheritedCallWeightAttr::parse(&content)?),
 			};
 			Ok(PalletAttr::RuntimeCall(attr, span))
-		} else if lookahead.peek(keyword::tasks_experimental) {
+		/* } else if lookahead.peek(keyword::tasks_experimental) {
 			Ok(PalletAttr::Tasks(content.parse::<keyword::tasks_experimental>()?.span()))
 		} else if lookahead.peek(keyword::task_enum) {
 			Ok(PalletAttr::RuntimeTask(content.parse::<keyword::task_enum>()?.span()))
@@ -772,7 +772,7 @@ impl syn::parse::Parse for PalletAttr {
 		} else if lookahead.peek(keyword::task_index) {
 			Ok(PalletAttr::TaskIndex(content.parse::<keyword::task_index>()?.span()))
 		} else if lookahead.peek(keyword::task_list) {
-			Ok(PalletAttr::TaskList(content.parse::<keyword::task_list>()?.span()))
+			Ok(PalletAttr::TaskList(content.parse::<keyword::task_list>()?.span())) */
 		} else if lookahead.peek(keyword::error) {
 			Ok(PalletAttr::Error(content.parse::<keyword::error>()?.span()))
 		} else if lookahead.peek(keyword::event) {
@@ -793,8 +793,8 @@ impl syn::parse::Parse for PalletAttr {
 			Ok(PalletAttr::TypeValue(content.parse::<keyword::type_value>()?.span()))
 		} else if lookahead.peek(keyword::extra_constants) {
 			Ok(PalletAttr::ExtraConstants(content.parse::<keyword::extra_constants>()?.span()))
-		} else if lookahead.peek(keyword::composite_enum) {
-			Ok(PalletAttr::Composite(content.parse::<keyword::composite_enum>()?.span()))
+		/* } else if lookahead.peek(keyword::composite_enum) {
+			Ok(PalletAttr::Composite(content.parse::<keyword::composite_enum>()?.span())) */
 		} else if lookahead.peek(keyword::view_functions) {
 			Ok(PalletAttr::ViewFunctions(content.parse::<keyword::view_functions>()?.span()))
 		} else {
diff --git a/substrate/frame/support/procedural/src/runtime/expand/mod.rs b/substrate/frame/support/procedural/src/runtime/expand/mod.rs
index ae466db..3571b36 100644
--- a/substrate/frame/support/procedural/src/runtime/expand/mod.rs
+++ b/substrate/frame/support/procedural/src/runtime/expand/mod.rs
@@ -177,11 +177,11 @@ fn construct_runtime_final_expansion(
 	let mut outer_event = None;
 	let mut outer_error = None;
 	let mut outer_origin = None;
-	let mut freeze_reason = None;
+	/* let mut freeze_reason = None;
 	let mut hold_reason = None;
 	let mut slash_reason = None;
 	let mut lock_id = None;
-	let mut task = None;
+	let mut task = None; */
 	let mut query = None;
 
 	for runtime_type in runtime_types.iter() {
@@ -210,7 +210,7 @@ fn construct_runtime_final_expansion(
 				outer_origin =
 					Some(expand::expand_outer_origin(&name, system_pallet, &pallets, &scrate)?);
 			},
-			RuntimeType::RuntimeFreezeReason(_) => {
+			/* RuntimeType::RuntimeFreezeReason(_) => {
 				freeze_reason = Some(expand::expand_outer_freeze_reason(&pallets, &scrate));
 			},
 			RuntimeType::RuntimeHoldReason(_) => {
@@ -224,7 +224,7 @@ fn construct_runtime_final_expansion(
 			},
 			RuntimeType::RuntimeTask(_) => {
 				task = Some(expand::expand_outer_task(&name, &pallets, &scrate));
-			},
+			}, */
 			RuntimeType::RuntimeViewFunction(_) => {
 				query = Some(expand::expand_outer_query(&name, &pallets, &scrate));
 			},
@@ -303,7 +303,7 @@ fn construct_runtime_final_expansion(
 
 		#dispatch
 
-		#task
+		// #task
 
 		#query
 
@@ -315,13 +315,13 @@ fn construct_runtime_final_expansion(
 
 		#validate_unsigned
 
-		#freeze_reason
+		/* #freeze_reason
 
 		#hold_reason
 
 		#lock_id
 
-		#slash_reason
+		#slash_reason */
 
 		#integrity_test
 
diff --git a/substrate/frame/support/procedural/src/runtime/parse/runtime_types.rs b/substrate/frame/support/procedural/src/runtime/parse/runtime_types.rs
index 9a38514..879c0ba 100644
--- a/substrate/frame/support/procedural/src/runtime/parse/runtime_types.rs
+++ b/substrate/frame/support/procedural/src/runtime/parse/runtime_types.rs
@@ -27,11 +27,11 @@ mod keyword {
 	custom_keyword!(RuntimeEvent);
 	custom_keyword!(RuntimeError);
 	custom_keyword!(RuntimeOrigin);
-	custom_keyword!(RuntimeFreezeReason);
+	/* custom_keyword!(RuntimeFreezeReason);
 	custom_keyword!(RuntimeHoldReason);
 	custom_keyword!(RuntimeSlashReason);
 	custom_keyword!(RuntimeLockId);
-	custom_keyword!(RuntimeTask);
+	custom_keyword!(RuntimeTask); */
 	custom_keyword!(RuntimeViewFunction);
 }
 
@@ -41,11 +41,11 @@ pub enum RuntimeType {
 	RuntimeEvent(keyword::RuntimeEvent),
 	RuntimeError(keyword::RuntimeError),
 	RuntimeOrigin(keyword::RuntimeOrigin),
-	RuntimeFreezeReason(keyword::RuntimeFreezeReason),
+	/* RuntimeFreezeReason(keyword::RuntimeFreezeReason),
 	RuntimeHoldReason(keyword::RuntimeHoldReason),
 	RuntimeSlashReason(keyword::RuntimeSlashReason),
 	RuntimeLockId(keyword::RuntimeLockId),
-	RuntimeTask(keyword::RuntimeTask),
+	RuntimeTask(keyword::RuntimeTask), */
 	RuntimeViewFunction(keyword::RuntimeViewFunction),
 }
 
@@ -61,7 +61,7 @@ impl Parse for RuntimeType {
 			Ok(Self::RuntimeError(input.parse()?))
 		} else if lookahead.peek(keyword::RuntimeOrigin) {
 			Ok(Self::RuntimeOrigin(input.parse()?))
-		} else if lookahead.peek(keyword::RuntimeFreezeReason) {
+		/* } else if lookahead.peek(keyword::RuntimeFreezeReason) {
 			Ok(Self::RuntimeFreezeReason(input.parse()?))
 		} else if lookahead.peek(keyword::RuntimeHoldReason) {
 			Ok(Self::RuntimeHoldReason(input.parse()?))
@@ -70,7 +70,7 @@ impl Parse for RuntimeType {
 		} else if lookahead.peek(keyword::RuntimeLockId) {
 			Ok(Self::RuntimeLockId(input.parse()?))
 		} else if lookahead.peek(keyword::RuntimeTask) {
-			Ok(Self::RuntimeTask(input.parse()?))
+			Ok(Self::RuntimeTask(input.parse()?)) */
 		} else if lookahead.peek(keyword::RuntimeViewFunction) {
 			Ok(Self::RuntimeViewFunction(input.parse()?))
 		} else {
diff --git a/substrate/frame/support/src/dispatch.rs b/substrate/frame/support/src/dispatch.rs
index 654e248..9544179 100644
--- a/substrate/frame/support/src/dispatch.rs
+++ b/substrate/frame/support/src/dispatch.rs
@@ -745,7 +745,7 @@ mod weight_tests {
 			type BaseCallFilter: crate::traits::Contains<Self::RuntimeCall>;
 			type RuntimeOrigin;
 			type RuntimeCall;
-			type RuntimeTask;
+			// type RuntimeTask;
 			type PalletInfo: crate::traits::PalletInfo;
 			type DbWeight: Get<crate::weights::RuntimeDbWeight>;
 		}
@@ -855,7 +855,7 @@ mod weight_tests {
 		type BaseCallFilter = crate::traits::Everything;
 		type RuntimeOrigin = RuntimeOrigin;
 		type RuntimeCall = RuntimeCall;
-		type RuntimeTask = RuntimeTask;
+		// type RuntimeTask = RuntimeTask;
 		type DbWeight = DbWeight;
 		type PalletInfo = PalletInfo;
 	}
@@ -1385,7 +1385,7 @@ mod extension_weight_tests {
 		type BaseCallFilter = crate::traits::Everything;
 		type RuntimeOrigin = RuntimeOrigin;
 		type RuntimeCall = RuntimeCall;
-		type RuntimeTask = RuntimeTask;
+		// type RuntimeTask = RuntimeTask;
 		type DbWeight = DbWeight;
 		type PalletInfo = PalletInfo;
 	}
diff --git a/substrate/frame/support/src/lib.rs b/substrate/frame/support/src/lib.rs
index 1921383..ac71cde 100644
--- a/substrate/frame/support/src/lib.rs
+++ b/substrate/frame/support/src/lib.rs
@@ -433,7 +433,7 @@ pub mod pallet_prelude {
 		traits::{
 			Authorize, BuildGenesisConfig, ConstU32, ConstUint, EnsureOrigin, Get, GetDefault,
 			GetStorageVersion, Hooks, IsType, OriginTrait, PalletInfoAccess, StorageInfoTrait,
-			StorageVersion, Task, TypedGet,
+			StorageVersion, TypedGet,
 		},
 		Blake2_128, Blake2_128Concat, Blake2_256, CloneNoBound, DebugNoBound, EqNoBound, Identity,
 		PartialEqNoBound, Twox128, Twox256, Twox64Concat,
@@ -1170,7 +1170,7 @@ pub mod pallet_macros {
 	/// ```
 	pub use frame_support_procedural::config;
 
-	/// Allows defining an enum that gets composed as an aggregate enum by `construct_runtime`.
+	/* /// Allows defining an enum that gets composed as an aggregate enum by `construct_runtime`.
 	///
 	/// The `#[pallet::composite_enum]` attribute allows you to define an enum that gets
 	/// composed as an aggregate enum by `construct_runtime`. This is similar in principle with
@@ -1218,5 +1218,5 @@ pub mod pallet_macros {
 	/// # 	#[pallet::config]
 	/// # 	pub trait Config: frame_system::Config {}
 	/// }
-	pub use frame_support_procedural::composite_enum;
+	pub use frame_support_procedural::composite_enum; */
 
@@ -2792,7 +2792,7 @@ pub mod pallet_macros {
 	pub use frame_support_procedural::storage;
 
 	pub use frame_support_procedural::{
-		authorize, task_condition, task_index, task_list, task_weight, tasks_experimental,
+		authorize, /* task_condition, task_index, task_list, task_weight, tasks_experimental, */
 		weight_of_authorize,
 	};
 
diff --git a/substrate/frame/system/src/lib.rs b/substrate/frame/system/src/lib.rs
index 9d80e6e..f90d3a6 100644
--- a/substrate/frame/system/src/lib.rs
+++ b/substrate/frame/system/src/lib.rs
@@ -362,8 +362,8 @@ pub mod pallet {
 			type RuntimeCall = ();
 			#[inject_runtime_type]
 			type PalletInfo = ();
-			#[inject_runtime_type]
-			type RuntimeTask = ();
+			/* #[inject_runtime_type]
+			type RuntimeTask = (); */
 			type BaseCallFilter = frame_support::traits::Everything;
 			type BlockHashCount = TestBlockHashCount<frame_support::traits::ConstU32<10>>;
 			type OnSetCode = ();
@@ -452,9 +452,9 @@ pub mod pallet {
 			#[inject_runtime_type]
 			type RuntimeCall = ();
 
-			/// The aggregated Task type, injected by `construct_runtime!`.
+			/* /// The aggregated Task type, injected by `construct_runtime!`.
 			#[inject_runtime_type]
-			type RuntimeTask = ();
+			type RuntimeTask = (); */
 
 			/// Converts a module to the index of the module, injected by `construct_runtime!`.
 			#[inject_runtime_type]
@@ -544,9 +544,9 @@ pub mod pallet {
 			+ From<Call<Self>>
 			+ Authorize;
 
-		/// The aggregated `RuntimeTask` type.
+		/* /// The aggregated `RuntimeTask` type.
 		#[pallet::no_default_bounds]
-		type RuntimeTask: Task;
+		type RuntimeTask: Task; */
 
 		/// This stores the number of previous transactions associated with a sender account.
 		type Nonce: Parameter
@@ -815,7 +815,7 @@ pub mod pallet {
 			Ok(().into())
 		}
 
-		#[cfg(feature = "experimental")]
+		/* #[cfg(feature = "experimental")]
 		#[pallet::call_index(8)]
 		#[pallet::weight(task.weight())]
 		pub fn do_task(_origin: OriginFor<T>, task: T::RuntimeTask) -> DispatchResultWithPostInfo {
@@ -834,7 +834,7 @@ pub mod pallet {
 
 			// Return success.
 			Ok(().into())
-		}
+		} */
 
 		/// Authorize an upgrade to a given `code_hash` for the runtime. The runtime can be supplied
 		/// later.
@@ -926,7 +926,7 @@ pub mod pallet {
 		KilledAccount { account: T::AccountId },
 		/// On on-chain remark happened.
 		Remarked { sender: T::AccountId, hash: T::Hash },
-		#[cfg(feature = "experimental")]
+		/* #[cfg(feature = "experimental")]
 		/// A [`Task`] has started executing
 		TaskStarted { task: T::RuntimeTask },
 		#[cfg(feature = "experimental")]
@@ -934,7 +934,7 @@ pub mod pallet {
 		TaskCompleted { task: T::RuntimeTask },
 		#[cfg(feature = "experimental")]
 		/// A [`Task`] failed during execution.
-		TaskFailed { task: T::RuntimeTask, err: DispatchError },
+		TaskFailed { task: T::RuntimeTask, err: DispatchError }, */
 		/// An upgrade was authorized.
 		UpgradeAuthorized { code_hash: T::Hash, check_version: bool },
 		/// An invalid authorized upgrade was rejected while trying to apply it.
@@ -962,12 +962,12 @@ pub mod pallet {
 		CallFiltered,
 		/// A multi-block migration is ongoing and prevents the current code from being replaced.
 		MultiBlockMigrationsOngoing,
-		#[cfg(feature = "experimental")]
+		/* #[cfg(feature = "experimental")]
 		/// The specified [`Task`] is not valid.
 		InvalidTask,
 		#[cfg(feature = "experimental")]
 		/// The specified [`Task`] failed during execution.
-		FailedTask,
+		FailedTask, */
 		/// No upgrade authorized.
 		NothingAuthorized,
 		/// The submitted code is not authorized.
@@ -1151,7 +1151,7 @@ pub mod pallet {
 				}
 			}
 
-			#[cfg(feature = "experimental")]
+			/* #[cfg(feature = "experimental")]
 			if let Call::do_task { ref task } = call {
 				// If valid, the tasks provides the tag: hash of task.
 				// But it is allowed to have many task for a single process, e.g. a task that takes
@@ -1171,9 +1171,9 @@ pub mod pallet {
 						});
 					}
 				}
 			}
 
 			#[cfg(not(feature = "experimental"))]
-			let _ = source;
+			let _ = source; */

 			Err(InvalidTransaction::Call.into())
