diff --git a/substrate/primitives/arithmetic/src/fixed_point.rs b/substrate/primitives/arithmetic/src/fixed_point.rs
index 9b5ef81..50dfdd5 100644
--- a/substrate/primitives/arithmetic/src/fixed_point.rs
+++ b/substrate/primitives/arithmetic/src/fixed_point.rs
@@ -26,10 +26,6 @@
 //! [`FixedU64`](FixedU64) (and its signed and `u128` counterparts) can be utilized.
 //! Take for example this very rudimentary pricing mechanism, where we wish to calculate the demand
 //! / supply to get a price for some on-chain compute:
-#![doc = docify::embed!(
-	"./src/lib.rs",
-	fixed_u64_block_computation_example
-)]
 //!
 //! For a much more comprehensive example, be sure to look at the source for broker (the "coretime")
 //! pallet.
@@ -38,10 +34,6 @@
 //!
 //! Just as with [`PerThing`](PerThing), you can also perform regular mathematical
 //! expressions:
-#![doc = docify::embed!(
-	"./src/lib.rs",
-	fixed_u64_operation_example
-)]
 //!
 
 use crate::{
