diff --git a/substrate/client/consensus/grandpa/src/finality_proof.rs b/substrate/client/consensus/grandpa/src/finality_proof.rs
index 8a47d12..1da18fb 100644
--- a/substrate/client/consensus/grandpa/src/finality_proof.rs
+++ b/substrate/client/consensus/grandpa/src/finality_proof.rs
@@ -437,8 +437,8 @@ mod tests {
 		auth: &[Ed25519Keyring],
 	) -> finality_grandpa::Commit<H256, u64, S, Id>
 	where
-		Id: From<sp_core::ed25519::Public>,
-		S: From<sp_core::ed25519::Signature>,
+		Id: From<sp_core::sr25519::Public>,
+		S: From<sp_core::sr25519::Signature>,
 	{
 		let mut precommits = Vec::new();
 
diff --git a/substrate/client/keystore/src/local.rs b/substrate/client/keystore/src/local.rs
index 180b167..0952f12 100644
--- a/substrate/client/keystore/src/local.rs
+++ b/substrate/client/keystore/src/local.rs
@@ -22,7 +22,7 @@ use parking_lot::RwLock;
 use sp_application_crypto::{AppCrypto, AppPair, IsWrappedBy};
 use sp_core::{
 	crypto::{ByteArray, ExposeSecret, KeyTypeId, Pair as CorePair, SecretString, VrfSecret},
-	ecdsa, ed25519, sr25519,
+	sr25519,
 };
 use sp_keystore::{Error as TraitError, Keystore, KeystorePtr};
 use std::{
@@ -212,6 +212,7 @@ impl Keystore for LocalKeystore {
 		self.vrf_pre_output::<sr25519::Pair>(key_type, public, input)
 	}
 
+	/*
 	fn ed25519_public_keys(&self, key_type: KeyTypeId) -> Vec<ed25519::Public> {
 		self.public_keys::<ed25519::Pair>(key_type)
 	}
@@ -273,6 +274,7 @@ impl Keystore for LocalKeystore {
 			.map(|pair| pair.sign_prehashed(msg));
 		Ok(sig)
 	}
+	*/
 
 	sp_keystore::bandersnatch_experimental_enabled! {
 		fn bandersnatch_public_keys(&self, key_type: KeyTypeId) -> Vec<bandersnatch::Public> {
diff --git a/substrate/primitives/consensus/grandpa/src/lib.rs b/substrate/primitives/consensus/grandpa/src/lib.rs
index 3aa4dc2..2d220e4 100644
--- a/substrate/primitives/consensus/grandpa/src/lib.rs
+++ b/substrate/primitives/consensus/grandpa/src/lib.rs
@@ -43,8 +43,8 @@ pub const RUNTIME_LOG_TARGET: &str = "runtime::grandpa";
 pub const KEY_TYPE: sp_core::crypto::KeyTypeId = sp_application_crypto::key_types::GRANDPA;
 
 mod app {
-	use sp_application_crypto::{app_crypto, ed25519, key_types::GRANDPA};
-	app_crypto!(ed25519, GRANDPA);
+	use sp_application_crypto::{app_crypto, sr25519, key_types::GRANDPA};
+	app_crypto!(sr25519, GRANDPA);
 }
 
 sp_application_crypto::with_pair! {
@@ -476,7 +476,7 @@ where
 
 	let encoded = localized_payload(round, set_id, &message);
 	let signature = keystore
-		.ed25519_sign(AuthorityId::ID, public.as_ref(), &encoded[..])
+		.sr25519_sign(AuthorityId::ID, public.as_ref(), &encoded[..])
 		.ok()
 		.flatten()?
 		.try_into()
diff --git a/substrate/primitives/core/src/crypto.rs b/substrate/primitives/core/src/crypto.rs
index 03be04c..04b3920 100644
--- a/substrate/primitives/core/src/crypto.rs
+++ b/substrate/primitives/core/src/crypto.rs
@@ -17,5 +17,5 @@
 
 //! Cryptographic utilities.
 
-use crate::{ed25519, sr25519, U256};
+use crate::{sr25519, U256};
 use alloc::{format, str, vec::Vec};
@@ -594,11 +594,13 @@ impl From<sr25519::Public> for AccountId32 {
 	}
 }
 
+/*
 impl From<ed25519::Public> for AccountId32 {
 	fn from(k: ed25519::Public) -> Self {
 		k.0.into()
 	}
 }
+*/
 
 #[cfg(feature = "std")]
 impl std::fmt::Display for AccountId32 {
diff --git a/substrate/primitives/keyring/src/lib.rs b/substrate/primitives/keyring/src/lib.rs
index 36e77da..6f421c1 100644
--- a/substrate/primitives/keyring/src/lib.rs
+++ b/substrate/primitives/keyring/src/lib.rs
@@ -25,16 +25,12 @@ use alloc::fmt;
 /// Test account crypto for sr25519.
 pub mod sr25519;
 
-/// Test account crypto for ed25519.
-pub mod ed25519;
-
 /// Test account crypto for bandersnatch.
 #[cfg(feature = "bandersnatch-experimental")]
 pub mod bandersnatch;
 
 #[cfg(feature = "bandersnatch-experimental")]
 pub use bandersnatch::Keyring as BandersnatchKeyring;
-pub use ed25519::Keyring as Ed25519Keyring;
 pub use sr25519::Keyring as Sr25519Keyring;
 
 #[derive(Debug)]
diff --git a/substrate/primitives/keyring/src/sr25519.rs b/substrate/primitives/keyring/src/sr25519.rs
index 5ff9056..ed83b90 100644
--- a/substrate/primitives/keyring/src/sr25519.rs
+++ b/substrate/primitives/keyring/src/sr25519.rs
@@ -149,12 +149,6 @@ impl From<Keyring> for &'static str {
 	}
 }
 
-impl From<Keyring> for sp_runtime::MultiSigner {
-	fn from(x: Keyring) -> Self {
-		sp_runtime::MultiSigner::Sr25519(x.into())
-	}
-}
-
 impl FromStr for Keyring {
 	type Err = ParseKeyringError;
 
diff --git a/substrate/primitives/keystore/src/lib.rs b/substrate/primitives/keystore/src/lib.rs
index 42ad2c6..53b016e 100644
--- a/substrate/primitives/keystore/src/lib.rs
+++ b/substrate/primitives/keystore/src/lib.rs
@@ -30,7 +30,7 @@ use sp_core::bandersnatch;
 use sp_core::{bls381, ecdsa_bls381};
 use sp_core::{
 	crypto::{ByteArray, CryptoTypeId, KeyTypeId},
-	ecdsa, ed25519, sr25519,
+	sr25519,
 };
 
 use alloc::{string::String, sync::Arc, vec::Vec};
@@ -120,6 +120,7 @@ pub trait Keystore: Send + Sync {
 		input: &sr25519::vrf::VrfInput,
 	) -> Result<Option<sr25519::vrf::VrfPreOutput>, Error>;
 
+	/*
 	/// Returns all ed25519 public keys for the given key type.
 	fn ed25519_public_keys(&self, key_type: KeyTypeId) -> Vec<ed25519::Public>;
 
@@ -190,6 +191,7 @@ pub trait Keystore: Send + Sync {
 		public: &ecdsa::Public,
 		msg: &[u8; 32],
 	) -> Result<Option<ecdsa::Signature>, Error>;
+	*/
 
 	/// Returns all the bandersnatch public keys for the given key type.
 	#[cfg(feature = "bandersnatch-experimental")]
@@ -401,6 +403,7 @@ pub trait Keystore: Send + Sync {
 					.map_err(|_| Error::ValidationError("Invalid public key format".into()))?;
 				self.sr25519_sign(id, &public, msg)?.map(|s| s.encode())
 			},
+			/*
 			ed25519::CRYPTO_ID => {
 				let public = ed25519::Public::from_slice(public)
 					.map_err(|_| Error::ValidationError("Invalid public key format".into()))?;
@@ -412,6 +415,7 @@ pub trait Keystore: Send + Sync {
 
 				self.ecdsa_sign(id, &public, msg)?.map(|s| s.encode())
 			},
+			*/
 			#[cfg(feature = "bandersnatch-experimental")]
 			bandersnatch::CRYPTO_ID => {
 				let public = bandersnatch::Public::from_slice(public)
@@ -476,6 +480,7 @@ impl<T: Keystore + ?Sized> Keystore for Arc<T> {
 		(**self).sr25519_vrf_pre_output(key_type, public, input)
 	}
 
+	/*
 	fn ed25519_public_keys(&self, key_type: KeyTypeId) -> Vec<ed25519::Public> {
 		(**self).ed25519_public_keys(key_type)
 	}
@@ -526,6 +531,7 @@ impl<T: Keystore + ?Sized> Keystore for Arc<T> {
 	) -> Result<Option<ecdsa::Signature>, Error> {
 		(**self).ecdsa_sign_prehashed(key_type, public, msg)
 	}
+	*/
 
 	#[cfg(feature = "bandersnatch-experimental")]
 	fn bandersnatch_public_keys(&self, key_type: KeyTypeId) -> Vec<bandersnatch::Public> {
diff --git a/substrate/primitives/keystore/src/testing.rs b/substrate/primitives/keystore/src/testing.rs
index 7939ee8..379a17b 100644
--- a/substrate/primitives/keystore/src/testing.rs
+++ b/substrate/primitives/keystore/src/testing.rs
@@ -27,4 +27,4 @@ use sp_core::bandersnatch;
 use sp_core::{
 	crypto::{ByteArray, KeyTypeId, Pair, VrfSecret},
-	ecdsa, ed25519, sr25519,
+	sr25519,
 };
@@ -181,5 +181,6 @@ impl Keystore for MemoryKeystore {
 	}
 
+	/*
 	fn ed25519_public_keys(&self, key_type: KeyTypeId) -> Vec<ed25519::Public> {
 		self.public_keys::<ed25519::Pair>(key_type)
 	}
@@ -378,3 +378,4 @@ impl Keystore for MemoryKeystore {
 	}
+	*/
 
 	fn insert(&self, key_type: KeyTypeId, suri: &str, public: &[u8]) -> Result<(), ()> {
diff --git a/substrate/primitives/runtime/src/lib.rs b/substrate/primitives/runtime/src/lib.rs
index b9f69b6..2f4c483 100644
--- a/substrate/primitives/runtime/src/lib.rs
+++ b/substrate/primitives/runtime/src/lib.rs
@@ -74,7 +74,6 @@ pub use sp_core::storage::{Storage, StorageChild};
 
 use sp_core::{
 	crypto::{self, ByteArray, FromEntropy},
-	ecdsa, ed25519,
 	hash::{H256, H512},
 	sr25519,
 };
@@ -270,6 +269,7 @@ impl BuildStorage for () {
 /// Consensus engine unique ID.
 pub type ConsensusEngineId = [u8; 4];
 
+/*
 /// Signature verify that can work with any known signature types.
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(
@@ -515,6 +513,7 @@ impl From<ed25519::Signature> for AnySignature {
 		Self(s.into())
 	}
 }
+*/
 
 impl From<DispatchError> for DispatchOutcome {
 	fn from(err: DispatchError) -> Self {
diff --git a/substrate/primitives/runtime/src/traits/mod.rs b/substrate/primitives/runtime/src/traits/mod.rs
index 0cda70e..af9978b 100644
--- a/substrate/primitives/runtime/src/traits/mod.rs
+++ b/substrate/primitives/runtime/src/traits/mod.rs
@@ -84,12 +84,14 @@ pub trait IdentifyAccount {
 	fn into_account(self) -> Self::AccountId;
 }
 
+/*
 impl IdentifyAccount for sp_core::ed25519::Public {
 	type AccountId = Self;
 	fn into_account(self) -> Self {
 		self
 	}
 }
+*/
 
 impl IdentifyAccount for sp_core::sr25519::Public {
 	type AccountId = Self;
@@ -98,10 +100,12 @@ impl IdentifyAccount for sp_core::sr25519::Public {
 	}
 }
 
+/*
 impl IdentifyAccount for sp_core::ecdsa::Public {
 	type AccountId = Self;
 	fn into_account(self) -> Self {
 		self
 	}
 }
+*/
 
@@ -119,6 +123,7 @@ pub trait Verify {
 	) -> bool;
 }
 
+/*
 impl Verify for sp_core::ed25519::Signature {
 	type Signer = sp_core::ed25519::Public;
 
@@ -126,6 +131,7 @@ impl Verify for sp_core::ed25519::Signature {
 		sp_io::crypto::ed25519_verify(self, msg.get(), signer)
 	}
 }
+*/
 
 impl Verify for sp_core::sr25519::Signature {
 	type Signer = sp_core::sr25519::Public;
@@ -135,6 +141,7 @@ impl Verify for sp_core::sr25519::Signature {
 	}
 }
 
+/*
 impl Verify for sp_core::ecdsa::Signature {
 	type Signer = sp_core::ecdsa::Public;
 	fn verify<L: Lazy<[u8]>>(&self, mut msg: L, signer: &sp_core::ecdsa::Public) -> bool {
@@ -147,6 +154,7 @@ impl Verify for sp_core::ecdsa::Signature {
 		<sp_core::ecdsa_bls381::Pair as sp_core::Pair>::verify(self, msg.get(), signer)
 	}
 }
+*/
 
 /// Means of signature verification of an application key.
 pub trait AppVerify {
diff --git a/substrate/primitives/io/src/lib.rs b/substrate/primitives/io/src/lib.rs
index e9f0dae5..455e99a1 100644
--- a/substrate/primitives/io/src/lib.rs
+++ b/substrate/primitives/io/src/lib.rs
@@ -99,6 +99,5 @@ use sp_core::{
 use sp_core::{
 	crypto::KeyTypeId,
-	ecdsa, ed25519,
 	offchain::{
 		HttpError, HttpRequestId, HttpRequestStatus, OpaqueNetworkState, StorageKind, Timestamp,
 	},
@@ -124,12 +123,6 @@ use sp_runtime_interface::{
 
 use codec::{Decode, Encode};
 
-#[cfg(not(substrate_runtime))]
-use secp256k1::{
-	ecdsa::{RecoverableSignature, RecoveryId},
-	Message,
-};
-
 #[cfg(not(substrate_runtime))]
 use sp_externalities::{Externalities, ExternalitiesExt};
 
@@ -148,17 +142,6 @@ mod global_alloc_riscv;
 #[cfg(not(substrate_runtime))]
 const LOG_TARGET: &str = "runtime::io";
 
-/// Error verifying ECDSA signature
-#[derive(Encode, Decode)]
-pub enum EcdsaVerifyError {
-	/// Incorrect value of R or S
-	BadRS,
-	/// Incorrect value of V
-	BadV,
-	/// Invalid signature
-	BadSignature,
-}
-
 /// The outcome of calling `storage_kill`. Returned value is the number of storage items
 /// removed from the backend from making the `storage_kill` call.
 #[derive(Encode, Decode)]
@@ -852,6 +834,7 @@ pub trait Misc {
 	}
 }
 
+/*
 #[cfg(not(substrate_runtime))]
 sp_externalities::decl_extension! {
 	/// Extension to signal to [`crypt::ed25519_verify`] to use the dalek crate.
@@ -879,10 +862,12 @@ impl Default for UseDalekExt {
 		Self
 	}
 }
+*/
 
 /// Interfaces for working with crypto related types from within the runtime.
 #[runtime_interface]
 pub trait Crypto {
+        /*
 	/// Returns all `ed25519` public keys for the given key id from the keystore.
 	fn ed25519_public_keys(
 		&mut self,
@@ -984,6 +969,7 @@ pub trait Crypto {
 
 		res
 	}
+        */
 
 	/// Verify `sr25519` signature.
 	///
@@ -1119,6 +1105,7 @@ pub trait Crypto {
 		sr25519::Pair::verify_deprecated(sig, msg, pubkey)
 	}
 
+        /*
 	/// Returns all `ecdsa` public keys for the given key id from the keystore.
 	fn ecdsa_public_keys(
 		&mut self,
@@ -1436,6 +1423,7 @@ pub trait Crypto {
 			.ok()
 			.flatten()
 	}
+        */
 }
 
 /// Interface that provides functions for hashing with different algorithms.
