diff --git a/substrate/primitives/arithmetic/src/traits.rs b/substrate/primitives/arithmetic/src/traits.rs
index 3e9a34e62b..82d0937147 100644
--- a/substrate/primitives/arithmetic/src/traits.rs
+++ b/substrate/primitives/arithmetic/src/traits.rs
@@ -27,6 +27,6 @@ pub use ensure::{
 	EnsureOpAssign, EnsureSub, EnsureSubAssign,
 };
-pub use integer_sqrt::IntegerSquareRoot;
+// pub use integer_sqrt::IntegerSquareRoot;
 pub use num_traits::{
 	checked_pow, Bounded, CheckedAdd, CheckedDiv, CheckedMul, CheckedNeg, CheckedRem, CheckedShl,
 	CheckedShr, CheckedSub, One, Signed, Unsigned, Zero,
@@ -39,7 +39,7 @@ pub trait BaseArithmetic:
 	From<u8>
 	+ Zero
 	+ One
-	+ IntegerSquareRoot
+	// + IntegerSquareRoot
 	+ Add<Self, Output = Self>
 	+ AddAssign<Self>
 	+ Sub<Self, Output = Self>
@@ -97,7 +97,7 @@ impl<
 		T: From<u8>
 			+ Zero
 			+ One
-			+ IntegerSquareRoot
+			// + IntegerSquareRoot
 			+ Add<Self, Output = Self>
 			+ AddAssign<Self>
 			+ Sub<Self, Output = Self>
diff --git a/substrate/primitives/runtime/src/traits/mod.rs b/substrate/primitives/runtime/src/traits/mod.rs
index 874bc53542..f60ea3a9d9 100644
--- a/substrate/primitives/runtime/src/traits/mod.rs
+++ b/substrate/primitives/runtime/src/traits/mod.rs
@@ -40,7 +40,7 @@ pub use sp_arithmetic::traits::{
 	checked_pow, ensure_pow, AtLeast32Bit, AtLeast32BitUnsigned, Bounded, CheckedAdd, CheckedDiv,
 	CheckedMul, CheckedShl, CheckedShr, CheckedSub, Ensure, EnsureAdd, EnsureAddAssign, EnsureDiv,
 	EnsureDivAssign, /* EnsureFixedPointNumber, */ EnsureFrom, EnsureInto, EnsureMul, EnsureMulAssign,
-	EnsureOp, EnsureOpAssign, EnsureSub, EnsureSubAssign, IntegerSquareRoot, One,
+	EnsureOp, EnsureOpAssign, EnsureSub, EnsureSubAssign, /* IntegerSquareRoot, */ One,
 	SaturatedConversion, Saturating, UniqueSaturatedFrom, UniqueSaturatedInto, Zero,
 };
 use sp_core::{self, storage::StateVersion, Hasher, TypeId, U256};
