diff --git a/substrate/client/service/src/builder.rs b/substrate/client/service/src/builder.rs
index 74d94c30..ab13a26f 100644
--- a/substrate/client/service/src/builder.rs
+++ b/substrate/client/service/src/builder.rs
@@ -53,6 +53,6 @@ use sc_network::{
 };
 use sc_network_common::role::{Role, Roles};
-use sc_network_light::light_client_requests::handler::LightClientRequestHandler;
+// use sc_network_light::light_client_requests::handler::LightClientRequestHandler;
 use sc_network_sync::{
 	block_relay_protocol::{BlockDownloader, BlockRelayParams},
 	block_request_handler::BlockRequestHandler,
@@ -1135,7 +1135,7 @@ where
 
 	let genesis_hash = client.info().genesis_hash;
 
-	let light_client_request_protocol_config = {
+	/* let light_client_request_protocol_config = {
 		// Allow both outgoing and incoming requests.
 		let (handler, protocol_config) =
 			LightClientRequestHandler::new::<Net>(&protocol_id, fork_id, client.clone());
@@ -1144,5 +1144,5 @@ where
 	};
 
 	// install request handlers to `FullNetworkConfiguration`
-	net_config.add_request_response_protocol(light_client_request_protocol_config);
+	net_config.add_request_response_protocol(light_client_request_protocol_config); */
 
