diff --git a/substrate/client/cli/src/params/keystore_params.rs b/substrate/client/cli/src/params/keystore_params.rs
index 5a5d0f9499..e657d889b7 100644
--- a/substrate/client/cli/src/params/keystore_params.rs
+++ b/substrate/client/cli/src/params/keystore_params.rs
@@ -35,9 +35,9 @@ pub struct KeystoreParams {
 	#[arg(long, value_name = "PATH")]
 	pub keystore_path: Option<PathBuf>,
 
-	/// Use interactive shell for entering the password used by the keystore.
+	/* /// Use interactive shell for entering the password used by the keystore.
 	#[arg(long, conflicts_with_all = &["password", "password_filename"])]
-	pub password_interactive: bool,
+	pub password_interactive: bool, */
 
 	/// Password used by the keystore.
 	///
@@ -45,7 +45,7 @@ pub struct KeystoreParams {
 	#[arg(
 		long,
 		value_parser = secret_string_from_str,
-		conflicts_with_all = &["password_interactive", "password_filename"]
+		conflicts_with_all = &["password_filename"]
 	)]
 	pub password: Option<SecretString>,
 
@@ -53,7 +53,7 @@ pub struct KeystoreParams {
 	#[arg(
 		long,
 		value_name = "PATH",
-		conflicts_with_all = &["password_interactive", "password"]
+		conflicts_with_all = &["password"]
 	)]
 	pub password_filename: Option<PathBuf>,
 }
@@ -66,9 +66,9 @@ pub fn secret_string_from_str(s: &str) -> std::result::Result<SecretString, Stri
 impl KeystoreParams {
 	/// Get the keystore configuration for the parameters
 	pub fn keystore_config(&self, config_dir: &Path) -> Result<KeystoreConfig> {
-		let password = if self.password_interactive {
+		let password = /* if self.password_interactive {
 			Some(SecretString::new(input_keystore_password()?))
-		} else if let Some(ref file) = self.password_filename {
+		} else */ if let Some(ref file) = self.password_filename {
 			let password = fs::read_to_string(file).map_err(|e| format!("{}", e))?;
 			Some(SecretString::new(password))
 		} else {
@@ -83,7 +83,7 @@ impl KeystoreParams {
 		Ok(KeystoreConfig::Path { path, password })
 	}
 
-	/// helper method to fetch password from `KeyParams` or read from stdin
+	/* /// helper method to fetch password from `KeyParams` or read from stdin
 	pub fn read_password(&self) -> error::Result<Option<SecretString>> {
 		let (password_interactive, password) = (self.password_interactive, self.password.clone());
 
@@ -95,9 +95,9 @@ impl KeystoreParams {
 		};
 
 		Ok(pass)
-	}
+	} */
 }
 
-fn input_keystore_password() -> Result<String> {
+/* fn input_keystore_password() -> Result<String> {
 	rpassword::prompt_password("Keystore password: ").map_err(|e| format!("{:?}", e).into())
-}
+} */
