diff --git a/substrate/client/service/src/builder.rs b/substrate/client/service/src/builder.rs
index 69c7619a700..f7f24b44a6e 100644
--- a/substrate/client/service/src/builder.rs
+++ b/substrate/client/service/src/builder.rs
@@ -67,20 +67,20 @@
 	SyncingService, WarpSyncConfig,
 };
 use sc_rpc::{
-	author::AuthorApiServer,
+       /* author::AuthorApiServer,
 	chain::ChainApiServer,
 	offchain::OffchainApiServer,
-	state::{ChildStateApiServer, StateApiServer},
+       state::{ChildStateApiServer, StateApiServer}, */
 	system::SystemApiServer,
 	DenyUnsafe, SubscriptionTaskExecutor,
 };
-use sc_rpc_spec_v2::{
+/* use sc_rpc_spec_v2::{
 	archive::ArchiveApiServer,
 	/* bitswap::BitswapApiServer, */
 	chain_head::ChainHeadApiServer,
 	chain_spec::ChainSpecApiServer,
 	transaction::{TransactionApiServer, TransactionBroadcastApiServer},
-};
+}; */
 use sc_telemetry::{telemetry, ConnectionMessage, Telemetry, TelemetryHandle, SUBSTRATE_INFO};
 use sc_tracing::block::TracingExecuteBlock;
 use sc_transaction_pool_api::{MaintainedTransactionPool, TransactionPool};
@@ -619,13 +619,13 @@ where
 	let rpc_id_provider = config.rpc.id_provider.take();
 
 	// jsonrpsee RPC
-	// RPC-V2 specific metrics need to be registered before the RPC server is started,
+	/* // RPC-V2 specific metrics need to be registered before the RPC server is started,
 	// since we might have two instances running (one for the in-memory RPC and one for the network
 	// RPC).
 	let rpc_v2_metrics = config
 		.prometheus_registry()
 		.map(|registry| sc_rpc_spec_v2::transaction::TransactionMetrics::new(registry))
-		.transpose()?;
+		.transpose()?; */
 
 	// Create dedicated RPC runtime with limited blocking threads.
 	// This isolates RPC blocking operations from the rest of the node.
@@ -651,7 +651,7 @@ where
 			blocks_pruning: config.blocks_pruning,
 			backend: backend.clone(),
 			rpc_builder: &*rpc_builder,
-			metrics: rpc_v2_metrics.clone(),
+			/* metrics: rpc_v2_metrics.clone(), */
 			sync_oracle: sync_service.clone(),
 			tracing_execute_block: execute_block.clone(),
 		})
@@ -818,8 +818,8 @@ pub struct GenRpcModuleParams<'a, TBl: BlockT, TBackend, TCl, TRpc, TExPool> {
 	pub backend: Arc<TBackend>,
 	/// RPC builder.
 	pub rpc_builder: &'a dyn Fn(SubscriptionTaskExecutor) -> Result<RpcModule<TRpc>, Error>,
-	/// Transaction metrics handle.
-	pub metrics: Option<sc_rpc_spec_v2::transaction::TransactionMetrics>,
+	/* /// Transaction metrics handle.
+	pub metrics: Option<sc_rpc_spec_v2::transaction::TransactionMetrics>, */
 	/// Sync oracle for determining sync status.
 	pub sync_oracle: Arc<dyn sp_consensus::SyncOracle + Send + Sync>,
 	/// Optional [`TracingExecuteBlock`] handle.
@@ -843,7 +843,7 @@ pub fn gen_rpc_module<TBl, TBackend, TCl, TRpc, TExPool>(
 		blocks_pruning,
 		backend,
 		rpc_builder,
-		metrics,
+		/* metrics, */
 		sync_oracle,
 		tracing_execute_block: execute_block,
 	}: GenRpcModuleParams<TBl, TBackend, TCl, TRpc, TExPool>,
@@ -868,18 +868,18 @@ where
 	TBl::Hash: Unpin,
 	TBl::Header: Unpin,
 {
-	let system_info = sc_rpc::system::SystemInfo {
+	/* let system_info = sc_rpc::system::SystemInfo {
 		chain_name: chain_spec.name().into(),
 		impl_name,
 		impl_version,
 		properties: chain_spec.properties(),
 		chain_type: chain_spec.chain_type(),
-	};
+	}; */
 
 	let mut rpc_api = RpcModule::new(());
 	let task_executor = spawn_handle;
 
-	let (chain, state, child_state) = {
+	/* let (chain, state, child_state) = {
 		let chain = sc_rpc::chain::new_full(client.clone(), task_executor.clone()).into_rpc();
 		let (state, child_state) =
 			sc_rpc::state::new_full(client.clone(), task_executor.clone(), execute_block);
@@ -975,7 +975,7 @@ where
 	rpc_api.merge(author).map_err(|e| Error::Application(e.into()))?;
 	rpc_api.merge(system).map_err(|e| Error::Application(e.into()))?;
 	rpc_api.merge(state).map_err(|e| Error::Application(e.into()))?;
-	rpc_api.merge(child_state).map_err(|e| Error::Application(e.into()))?;
+	rpc_api.merge(child_state).map_err(|e| Error::Application(e.into()))?; */
 	// Additional [`RpcModule`]s defined in the node to fit the specific blockchain
 	let extra_rpcs = rpc_builder(task_executor.clone())?;
 	rpc_api.merge(extra_rpcs).map_err(|e| Error::Application(e.into()))?;
