diff --git a/substrate/client/service/src/builder.rs b/substrate/client/service/src/builder.rs
index 1ec349fc55..3fc52347e7 100644
--- a/substrate/client/service/src/builder.rs
+++ b/substrate/client/service/src/builder.rs
@@ -67,19 +67,19 @@ use sc_network_sync::{
 	SyncingService, WarpSyncConfig,
 };
 use sc_rpc::{
-	author::AuthorApiServer,
+	/* author::AuthorApiServer,
 	chain::ChainApiServer,
 	offchain::OffchainApiServer,
 	state::{ChildStateApiServer, StateApiServer},
-	system::SystemApiServer,
+	system::SystemApiServer, */
 	DenyUnsafe, SubscriptionTaskExecutor,
 };
-use sc_rpc_spec_v2::{
+/* use sc_rpc_spec_v2::{
 	archive::ArchiveApiServer,
 	chain_head::ChainHeadApiServer,
 	chain_spec::ChainSpecApiServer,
 	transaction::{TransactionApiServer, TransactionBroadcastApiServer},
-};
+}; */
 use sc_telemetry::{telemetry, ConnectionMessage, Telemetry, TelemetryHandle, SUBSTRATE_INFO};
 use sc_tracing::block::TracingExecuteBlock;
 use sc_transaction_pool_api::{MaintainedTransactionPool, TransactionPool};
@@ -598,13 +598,13 @@ where
 	let rpc_id_provider = config.rpc.id_provider.take();
 
 	// jsonrpsee RPC
-	// RPC-V2 specific metrics need to be registered before the RPC server is started,
+	/* // RPC-V2 specific metrics need to be registered before the RPC server is started,
 	// since we might have two instances running (one for the in-memory RPC and one for the network
 	// RPC).
 	let rpc_v2_metrics = config
 		.prometheus_registry()
 		.map(|registry| sc_rpc_spec_v2::transaction::TransactionMetrics::new(registry))
-		.transpose()?;
+		.transpose()?; */
 
 	let gen_rpc_module = || {
 		gen_rpc_module(GenRpcModuleParams {
@@ -620,7 +620,7 @@ where
 			blocks_pruning: config.blocks_pruning,
 			backend: backend.clone(),
 			rpc_builder: &*rpc_builder,
-			metrics: rpc_v2_metrics.clone(),
+			// metrics: rpc_v2_metrics.clone(),
 			tracing_execute_block: execute_block.clone(),
 		})
 	};
@@ -781,8 +781,8 @@ pub struct GenRpcModuleParams<'a, TBl: BlockT, TBackend, TCl, TRpc, TExPool> {
 	pub backend: Arc<TBackend>,
 	/// RPC builder.
 	pub rpc_builder: &'a dyn Fn(SubscriptionTaskExecutor) -> Result<RpcModule<TRpc>, Error>,
-	/// Transaction metrics handle.
-	pub metrics: Option<sc_rpc_spec_v2::transaction::TransactionMetrics>,
+	/* /// Transaction metrics handle.
+	pub metrics: Option<sc_rpc_spec_v2::transaction::TransactionMetrics>, */
 	/// Optional [`TracingExecuteBlock`] handle.
 	///
 	/// Will be used by the `trace_block` RPC to execute the actual block.
@@ -804,7 +804,7 @@ pub fn gen_rpc_module<TBl, TBackend, TCl, TRpc, TExPool>(
 		blocks_pruning,
 		backend,
 		rpc_builder,
-		metrics,
+		// metrics,
 		tracing_execute_block: execute_block,
 	}: GenRpcModuleParams<TBl, TBackend, TCl, TRpc, TExPool>,
 ) -> Result<RpcModule<()>, Error>
@@ -828,18 +828,18 @@ where
 	TBl::Hash: Unpin,
 	TBl::Header: Unpin,
 {
-	let system_info = sc_rpc::system::SystemInfo {
+	/* let system_info = sc_rpc::system::SystemInfo {
 		chain_name: chain_spec.name().into(),
 		impl_name,
 		impl_version,
 		properties: chain_spec.properties(),
 		chain_type: chain_spec.chain_type(),
-	};
+	}; */
 
 	let mut rpc_api = RpcModule::new(());
 	let task_executor = Arc::new(spawn_handle);
 
-	let (chain, state, child_state) = {
+	/* let (chain, state, child_state) = {
 		let chain = sc_rpc::chain::new_full(client.clone(), task_executor.clone()).into_rpc();
 		let (state, child_state) =
 			sc_rpc::state::new_full(client.clone(), task_executor.clone(), execute_block);
@@ -931,7 +931,7 @@ where
 	rpc_api.merge(author).map_err(|e| Error::Application(e.into()))?;
 	rpc_api.merge(system).map_err(|e| Error::Application(e.into()))?;
 	rpc_api.merge(state).map_err(|e| Error::Application(e.into()))?;
-	rpc_api.merge(child_state).map_err(|e| Error::Application(e.into()))?;
+	rpc_api.merge(child_state).map_err(|e| Error::Application(e.into()))?; */
 	// Additional [`RpcModule`]s defined in the node to fit the specific blockchain
 	let extra_rpcs = rpc_builder(task_executor.clone())?;
 	rpc_api.merge(extra_rpcs).map_err(|e| Error::Application(e.into()))?;
