diff --git a/substrate/client/rpc-api/src/author/mod.rs b/substrate/client/rpc-api/src/author/mod.rs
index c39d6c6..aa26480 100644
--- a/substrate/client/rpc-api/src/author/mod.rs
+++ b/substrate/client/rpc-api/src/author/mod.rs
@@ -37,7 +37,7 @@ pub trait AuthorApi<Hash, BlockHash> {
 	#[method(name = "author_insertKey", with_extensions)]
 	fn insert_key(&self, key_type: String, suri: String, public: Bytes) -> Result<(), Error>;
 
-	/// Generate new session keys and returns the corresponding public keys.
+	/* /// Generate new session keys and returns the corresponding public keys.
 	#[method(name = "author_rotateKeys", with_extensions)]
 	fn rotate_keys(&self) -> Result<Bytes, Error>;
 
@@ -47,7 +47,7 @@ pub trait AuthorApi<Hash, BlockHash> {
 	///
 	/// Returns `true` iff all private keys could be found.
 	#[method(name = "author_hasSessionKeys", with_extensions)]
-	fn has_session_keys(&self, session_keys: Bytes) -> Result<bool, Error>;
+	fn has_session_keys(&self, session_keys: Bytes) -> Result<bool, Error>; */
 
 	/// Checks if the keystore has private keys for the given public key and key type.
 	///
diff --git a/substrate/client/rpc/src/author/mod.rs b/substrate/client/rpc/src/author/mod.rs
index 0c99da1..4089dc7 100644
--- a/substrate/client/rpc/src/author/mod.rs
+++ b/substrate/client/rpc/src/author/mod.rs
@@ -38,7 +38,7 @@ use sp_blockchain::HeaderBackend;
 use sp_core::Bytes;
 use sp_keystore::{KeystoreExt, KeystorePtr};
 use sp_runtime::traits::Block as BlockT;
-use sp_session::SessionKeys;
+// use sp_session::SessionKeys;
 use std::sync::Arc;
 
 /// Re-export the API for backward compatibility.
@@ -80,7 +80,7 @@ impl<P, Client> AuthorApiServer<TxHash<P>, BlockHash<P>> for Author<P, Client>
 where
 	P: TransactionPool + Sync + Send + 'static,
 	Client: HeaderBackend<P::Block> + ProvideRuntimeApi<P::Block> + Send + Sync + 'static,
-	Client::Api: SessionKeys<P::Block>,
+	// Client::Api: SessionKeys<P::Block>,
 	P::Hash: Unpin,
 	<P::Block as BlockT>::Hash: Unpin,
 {
@@ -114,7 +114,7 @@ where
 		Ok(())
 	}
 
-	fn rotate_keys(&self, ext: &Extensions) -> Result<Bytes> {
+	/* fn rotate_keys(&self, ext: &Extensions) -> Result<Bytes> {
 		check_if_safe(ext)?;
 
 		let best_block_hash = self.client.info().best_hash;
@@ -140,7 +140,7 @@ where
 			.ok_or(Error::InvalidSessionKeys)?;
 
 		Ok(self.keystore.has_keys(&keys))
-	}
+	} */
 
 	fn has_key(&self, ext: &Extensions, public_key: Bytes, key_type: String) -> Result<bool> {
 		check_if_safe(ext)?;
diff --git a/substrate/client/service/src/builder.rs b/substrate/client/service/src/builder.rs
index 3919fb2..9be3226 100644
--- a/substrate/client/service/src/builder.rs
+++ b/substrate/client/service/src/builder.rs
@@ -473,7 +473,7 @@ where
 		+ 'static,
 	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_api::Core<TBl>
 		+ sp_transaction_pool::runtime_api::TaggedTransactionQueue<TBl>
-		+ sp_session::SessionKeys<TBl>
+		// + sp_session::SessionKeys<TBl>
 		+ sp_api::ApiExt<TBl>,
 	TBl: BlockT,
 	TBl::Hash: Unpin,
@@ -498,13 +498,13 @@ where
 
 	let chain_info = client.usage_info().chain;
 
-	sp_session::generate_initial_session_keys(
+	/* sp_session::generate_initial_session_keys(
 		client.clone(),
 		chain_info.best_hash,
 		config.dev_key_seed.clone().map(|s| vec![s]).unwrap_or_default(),
 		keystore.clone(),
 	)
-	.map_err(|e| Error::Application(Box::new(e)))?;
+	.map_err(|e| Error::Application(Box::new(e)))?; */
 
 	let sysinfo = sc_sysinfo::gather_sysinfo();
 	sc_sysinfo::print_sysinfo(&sysinfo);
@@ -770,7 +770,7 @@ where
 		+ Sync
 		+ 'static,
 	TBackend: sc_client_api::backend::Backend<TBl> + 'static,
-	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_session::SessionKeys<TBl> + sp_api::Core<TBl>,
+	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_api::Core<TBl>,
 	TExPool: MaintainedTransactionPool<Block = TBl, Hash = <TBl as BlockT>::Hash> + 'static,
 	TBl::Hash: Unpin,
 	TBl::Header: Unpin,
diff --git a/substrate/primitives/session/src/lib.rs b/substrate/primitives/session/src/lib.rs
index fbd39ef..8f5c27f 100644
--- a/substrate/primitives/session/src/lib.rs
+++ b/substrate/primitives/session/src/lib.rs
@@ -101,7 +101,7 @@ impl GetValidatorCount for MembershipProof {
 	}
 }
 
-/// Generate the initial session keys with the given seeds, at the given block and store them in
+/* /// Generate the initial session keys with the given seeds, at the given block and store them in
 /// the client's keystore.
 #[cfg(feature = "std")]
 pub fn generate_initial_session_keys<Block, T>(
@@ -130,4 +130,4 @@ where
 	}
 
 	Ok(())
-}
+} */
diff --git a/substrate/frame/src/lib.rs b/substrate/frame/src/lib.rs
index 9a50a95..2a4b350 100644
--- a/substrate/frame/src/lib.rs
+++ b/substrate/frame/src/lib.rs
@@ -435,7 +435,7 @@ pub mod runtime {
 		pub use sp_consensus_grandpa::*;
 		pub use sp_genesis_builder::*;
 		pub use sp_offchain::*;
-		pub use sp_session::runtime_api::*;
+		// pub use sp_session::runtime_api::*;
 		pub use sp_transaction_pool::runtime_api::*;
 	}
 
