diff --git a/substrate/client/service/src/builder.rs b/substrate/client/service/src/builder.rs
index 3919fb2..9be3226 100644
--- a/substrate/client/service/src/builder.rs
+++ b/substrate/client/service/src/builder.rs
@@ -473,7 +473,7 @@ where
 		+ 'static,
 	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_api::Core<TBl>
 		+ sp_transaction_pool::runtime_api::TaggedTransactionQueue<TBl>
-		+ sp_session::SessionKeys<TBl>
+		// + sp_session::SessionKeys<TBl>
 		+ sp_api::ApiExt<TBl>,
 	TBl: BlockT,
 	TBl::Hash: Unpin,
@@ -498,13 +498,13 @@ where
 
 	let chain_info = client.usage_info().chain;
 
-	sp_session::generate_initial_session_keys(
+	/* sp_session::generate_initial_session_keys(
 		client.clone(),
 		chain_info.best_hash,
 		config.dev_key_seed.clone().map(|s| vec![s]).unwrap_or_default(),
 		keystore.clone(),
 	)
-	.map_err(|e| Error::Application(Box::new(e)))?;
+	.map_err(|e| Error::Application(Box::new(e)))?; */
 
 	let sysinfo = sc_sysinfo::gather_sysinfo();
 	sc_sysinfo::print_sysinfo(&sysinfo);
@@ -770,7 +770,7 @@ where
 		+ Sync
 		+ 'static,
 	TBackend: sc_client_api::backend::Backend<TBl> + 'static,
-	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_session::SessionKeys<TBl> + sp_api::Core<TBl>,
+	<TCl as ProvideRuntimeApi<TBl>>::Api: sp_api::Core<TBl>,
 	TExPool: MaintainedTransactionPool<Block = TBl, Hash = <TBl as BlockT>::Hash> + 'static,
 	TBl::Hash: Unpin,
 	TBl::Header: Unpin,
diff --git a/substrate/primitives/session/src/lib.rs b/substrate/primitives/session/src/lib.rs
index fbd39ef..8f5c27f 100644
--- a/substrate/primitives/session/src/lib.rs
+++ b/substrate/primitives/session/src/lib.rs
@@ -101,7 +101,7 @@ impl GetValidatorCount for MembershipProof {
 	}
 }
 
-/// Generate the initial session keys with the given seeds, at the given block and store them in
+/* /// Generate the initial session keys with the given seeds, at the given block and store them in
 /// the client's keystore.
 #[cfg(feature = "std")]
 pub fn generate_initial_session_keys<Block, T>(
@@ -130,4 +130,4 @@ where
 	}
 
 	Ok(())
-}
+} */
 
