diff --git a/substrate/client/executor/wasmtime/Cargo.toml b/substrate/client/executor/wasmtime/Cargo.toml
index 976779631e..423f989452 100644
--- a/substrate/client/executor/wasmtime/Cargo.toml
+++ b/substrate/client/executor/wasmtime/Cargo.toml
@@ -31,7 +31,7 @@ sp-runtime-interface.workspace = true
 sp-wasm-interface = { features = ["wasmtime"], workspace = true, default-features = true }
 wasmtime = { features = [
 	"addr2line",
-	"cache",
+	# "cache",
 	"cranelift",
 	"demangle",
 	"gc",
diff --git a/substrate/client/executor/wasmtime/src/runtime.rs b/substrate/client/executor/wasmtime/src/runtime.rs
index 2e47f14693..bba385a293 100644
--- a/substrate/client/executor/wasmtime/src/runtime.rs
+++ b/substrate/client/executor/wasmtime/src/runtime.rs
@@ -41,7 +41,7 @@ use std::{
 		Arc,
 	},
 };
-use wasmtime::{AsContext, Cache, CacheConfig, Engine, Memory};
+use wasmtime::{AsContext, /* Cache, CacheConfig,*/ Engine, Memory};
 
 const MAX_INSTANCE_COUNT: u32 = 64;
 
@@ -191,7 +191,7 @@ impl WasmInstance for WasmtimeInstance {
 	}
 }
 
-/// Prepare a directory structure and a config file to enable wasmtime caching.
+/* /// Prepare a directory structure and a config file to enable wasmtime caching.
 ///
 /// In case of an error the caching will not be enabled.
 fn setup_wasmtime_caching(
@@ -213,7 +213,7 @@ fn setup_wasmtime_caching(
 	config.cache(Some(cache));
 
 	Ok(())
-}
+} */
 
 fn common_config(semantics: &Semantics) -> std::result::Result<wasmtime::Config, WasmError> {
 	let mut config = wasmtime::Config::new();
@@ -558,14 +558,14 @@ where
 	replace_strategy_if_broken(&mut config.semantics.instantiation_strategy);
 
 	let mut wasmtime_config = common_config(&config.semantics)?;
-	if let Some(ref cache_path) = config.cache_path {
+	/* if let Some(ref cache_path) = config.cache_path {
 		if let Err(reason) = setup_wasmtime_caching(cache_path, &mut wasmtime_config) {
 			log::warn!(
 				"failed to setup wasmtime cache. Performance may degrade significantly: {}.",
 				reason,
 			);
 		}
-	}
+	} */
 
 	let engine = Engine::new(&wasmtime_config)
 		.map_err(|e| WasmError::Other(format!("cannot create the wasmtime engine: {:#}", e)))?;
