diff --git a/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs b/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
index d8af26bf..48d46293 100644
--- a/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
+++ b/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
@@ -21,7 +21,7 @@
-use wasm_instrument::parity_wasm::elements::{
+/* use wasm_instrument::parity_wasm::elements::{
 	deserialize_buffer, serialize, ExportEntry, External, Internal, MemorySection, MemoryType,
 	Module, Section,
-};
+}; */
 
 /// A program blob containing a Substrate runtime.
 #[derive(Clone)]
@@ -29,7 +29,7 @@ pub struct RuntimeBlob(BlobKind);
 
 #[derive(Clone)]
 enum BlobKind {
-	WebAssembly(Module),
+	WebAssembly(Vec<u8>),
 	#[cfg(feature = "polkavm")] PolkaVM((polkavm::ProgramBlob, ArcBytes)),
 }
 
@@ -61,9 +61,9 @@ impl RuntimeBlob {
 			}
 		}
 
-		let raw_module: Module = deserialize_buffer(raw_blob)
-			.map_err(|e| WasmError::Other(format!("cannot deserialize module: {:?}", e)))?;
-		Ok(Self(BlobKind::WebAssembly(raw_module)))
+		/* let raw_module: Module = deserialize_buffer(raw_blob)
+			.map_err(|e| WasmError::Other(format!("cannot deserialize module: {:?}", e)))?; */
+		Ok(Self(BlobKind::WebAssembly(raw_blob.to_vec())))
 	}
 
 	/// Run a pass that instrument this module so as to introduce a deterministic stack height
@@ -79,13 +79,13 @@ impl RuntimeBlob {
 	///
 	/// Only valid for WASM programs; will return an error if the blob is a PolkaVM program.
 	pub fn inject_stack_depth_metering(self, stack_depth_limit: u32) -> Result<Self, WasmError> {
-		let injected_module =
+		/* let injected_module =
 			wasm_instrument::inject_stack_limiter(self.into_webassembly_blob()?, stack_depth_limit)
 				.map_err(|e| {
 					WasmError::Other(format!("cannot inject the stack limiter: {:?}", e))
-				})?;
+				})?; */
 
-		Ok(Self(BlobKind::WebAssembly(injected_module)))
+		Ok(Self(BlobKind::WebAssembly(self.into_webassembly_blob()?)))
 	}
 
 	/// Converts a WASM memory import into a memory section and exports it.
@@ -96,7 +96,7 @@ impl RuntimeBlob {
 	///
 	/// Only valid for WASM programs; will return an error if the blob is a PolkaVM program.
 	pub fn convert_memory_import_into_export(&mut self) -> Result<(), WasmError> {
-		let raw_module = self.as_webassembly_blob_mut()?;
+		/* let raw_module = self.as_webassembly_blob_mut()?;
 		let import_section = match raw_module.import_section_mut() {
 			Some(import_section) => import_section,
 			None => return Ok(()),
@@ -136,7 +136,7 @@ impl RuntimeBlob {
 				.push(ExportEntry::new(memory_name, Internal::Memory(0)));
 
 			break;
-		}
+		} */
 
 		Ok(())
 	}
@@ -151,7 +151,7 @@ impl RuntimeBlob {
 		&mut self,
 		heap_alloc_strategy: HeapAllocStrategy,
 	) -> Result<(), WasmError> {
-		let raw_module = self.as_webassembly_blob_mut()?;
+		/* let raw_module = self.as_webassembly_blob_mut()?;
 		let memory_section = raw_module
 			.memory_section_mut()
 			.ok_or_else(|| WasmError::Other("no memory section found".into()))?;
@@ -172,11 +172,11 @@ impl RuntimeBlob {
 				},
 			};
 			*memory_ty = MemoryType::new(min, max);
-		}
+		} */
 		Ok(())
 	}
 
-	/// Scans the wasm blob for the first section with the name that matches the given. Returns the
+	/* /// Scans the wasm blob for the first section with the name that matches the given. Returns the
 	/// contents of the custom section if found or `None` otherwise.
 	///
 	/// Only valid for WASM programs; will return an error if the blob is a PolkaVM program.
@@ -186,22 +186,22 @@ impl RuntimeBlob {
 			.custom_sections()
 			.find(|cs| cs.name() == section_name)
 			.map(|cs| cs.payload())
-	}
+	} */
 
 	/// Consumes this runtime blob and serializes it.
 	pub fn serialize(self) -> Vec<u8> {
 		match self.0 {
 			BlobKind::WebAssembly(raw_module) => {
-				serialize(raw_module).expect("serializing into a vec should succeed; qed")
+				raw_module // serialize(raw_module).expect("serializing into a vec should succeed; qed")
 			},
 			#[cfg(feature = "polkavm")] BlobKind::PolkaVM(ref blob) => blob.1.to_vec(),
 		}
 	}
 
-	fn as_webassembly_blob(&self) -> Result<&Module, WasmError> {
+	/* fn as_webassembly_blob(&self) -> Result<&Module, WasmError> {
 		match self.0 {
 			BlobKind::WebAssembly(ref raw_module) => Ok(raw_module),
 			#[cfg(feature = "polkavm")] BlobKind::PolkaVM(..) => Err(WasmError::Other(
 				"expected a WebAssembly program; found a PolkaVM program blob".into(),
 			)),
 		}
@@ -210,17 +210,17 @@ impl RuntimeBlob {
 	fn as_webassembly_blob_mut(&mut self) -> Result<&mut Module, WasmError> {
 		match self.0 {
 			BlobKind::WebAssembly(ref mut raw_module) => Ok(raw_module),
 			#[cfg(feature = "polkavm")] BlobKind::PolkaVM(..) => Err(WasmError::Other(
 				"expected a WebAssembly program; found a PolkaVM program blob".into(),
 			)),
 		}
-	}
+	} */
 
-	fn into_webassembly_blob(self) -> Result<Module, WasmError> {
+	fn into_webassembly_blob(self) -> Result<Vec<u8>, WasmError> {
 		match self.0 {
 			BlobKind::WebAssembly(raw_module) => Ok(raw_module),
 			#[cfg(feature = "polkavm")] BlobKind::PolkaVM(..) => Err(WasmError::Other(
 				"expected a WebAssembly program; found a PolkaVM program blob".into(),
 			)),
 		}
 	}
diff --git a/substrate/client/executor/src/wasm_runtime.rs b/substrate/client/executor/src/wasm_runtime.rs
index 8f189ca..97a79a1 100644
--- a/substrate/client/executor/src/wasm_runtime.rs
+++ b/substrate/client/executor/src/wasm_runtime.rs
@@ -353,7 +353,7 @@ fn decode_runtime_apis(apis: &[u8]) -> Result<Vec<([u8; 8], u32)>, WasmError> {
 /// If there are no such sections, it returns `None`. If there is an error during decoding those
 /// sections, `Err` will be returned.
 pub fn read_embedded_version(blob: &RuntimeBlob) -> Result<Option<RuntimeVersion>, WasmError> {
-	if let Some(mut version_section) = blob.custom_section_contents("runtime_version") {
+	/* if let Some(mut version_section) = blob.custom_section_contents("runtime_version") {
 		let apis = blob
 			.custom_section_contents("runtime_apis")
 			.map(decode_runtime_apis)
@@ -378,7 +378,7 @@ pub fn read_embedded_version(blob: &RuntimeBlob) -> Result<Option<RuntimeVersion
 		Ok(Some(decoded_version))
 	} else {
 		Ok(None)
-	}
+	} */ Ok(None)
 }
 
 fn create_versioned_wasm_runtime<H>(
diff --git a/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs b/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
index f14e7d52efb..9db25d541e6 100644
--- a/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
+++ b/substrate/client/executor/common/src/runtime_blob/runtime_blob.rs
@@ -142,6 +142,6 @@ impl RuntimeBlob {
 	}
 
-	/// Removes the maximum page limit from the memory section.
+	/* /// Removes the maximum page limit from the memory section.
 	///
 	/// Only valid for WASM programs; will return an error if the blob is a PolkaVM program.
 	pub fn clear_memory_max_limit(&mut self) -> Result<(), WasmError> {
@@ -154,7 +154,7 @@ impl RuntimeBlob {
 			*memory_ty = MemoryType::new(memory_ty.limits().initial(), None);
 		}
 		Ok(())
-	}
+	} */
 
 	/// Modifies the blob's memory section according to the given `heap_alloc_strategy`.
 	///
diff --git a/substrate/client/executor/wasmtime/src/runtime.rs b/substrate/client/executor/wasmtime/src/runtime.rs
index 49a96a61c9d..ae7b34a1f46 100644
--- a/substrate/client/executor/wasmtime/src/runtime.rs
+++ b/substrate/client/executor/wasmtime/src/runtime.rs
@@ -657,7 +657,7 @@ fn prepare_blob_for_compilation(
 	// now automatically take care of creating the memory for us, and it is also necessary
 	// to enable `wasmtime`'s instance pooling. (Imported memories are ineligible for pooling.)
 	blob.convert_memory_import_into_export()?;
-	blob.clear_memory_max_limit()?;
+	// blob.clear_memory_max_limit()?;
 
 	Ok(blob)
 }
