diff --git a/substrate/primitives/io/Cargo.toml b/substrate/primitives/io/Cargo.toml
index e440cc14..4448bed1 100644
--- a/substrate/primitives/io/Cargo.toml
+++ b/substrate/primitives/io/Cargo.toml
@@ -49,7 +49,7 @@ ed25519-dalek = { workspace = true }
 docify = { workspace = true }
 
 [target.'cfg(all(any(target_arch = "riscv32", target_arch = "riscv64"), substrate_runtime))'.dependencies]
-polkavm-derive = { workspace = true }
+polkavm-derive = { workspace = true, optional = true }
 
 [build-dependencies]
 rustversion = { workspace = true }
diff --git a/substrate/primitives/runtime-interface/Cargo.toml b/substrate/primitives/runtime-interface/Cargo.toml
index 9dbc7cef..acb415f8 100644
--- a/substrate/primitives/runtime-interface/Cargo.toml
+++ b/substrate/primitives/runtime-interface/Cargo.toml
@@ -29,7 +29,7 @@ sp-wasm-interface = { workspace = true }
 static_assertions = { workspace = true, default-features = true }
 
 [target.'cfg(all(target_arch = "riscv64", substrate_runtime))'.dependencies]
-polkavm-derive = { workspace = true }
+polkavm-derive = { workspace = true, optional = true }
 
 [dev-dependencies]
 rustversion = { workspace = true }
